// ionic-angular/components/scroll/scroll.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/scroll/scroll';
const styles_Scroll:any[] = ([] as any[]);
export const RenderType_Scroll:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Scroll,
  data: {}
}
);
export function View_Scroll_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(402653184,1,{_scrollContent: 0}),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'scrollContent',
        1
      ]

    ]
      ,(null as any),2,'div',[[
        'class',
        'scroll-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'scroll-zoom-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0)
  ]
  ,(null as any),(null as any));
}
function View_Scroll_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-scroll',([] as any[]),[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Scroll_0,RenderType_Scroll)),
    import0.ɵdid(49152,(null as any),0,import1.Scroll,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).scrollX;
    const currVal_1:any = import0.ɵnov(v,1).scrollY;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const ScrollNgFactory:import0.ComponentFactory<import1.Scroll> = import0.ɵccf('ion-scroll',import1.Scroll,View_Scroll_Host_0,{
  scrollX: 'scrollX',
  scrollY: 'scrollY',
  zoom: 'zoom',
  maxZoom: 'maxZoom'
}
,{},['*']);
