// ionic-angular/components/picker/picker-column.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/picker/picker-column';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/platform/platform';
import * as import5 from 'ionic-angular/tap-click/haptic';
import * as import6 from 'ionic-angular/platform/dom-controller';
const styles_PickerColumnCmp:any[] = ([] as any[]);
export const RenderType_PickerColumnCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PickerColumnCmp,
  data: {}
}
);
function View_PickerColumnCmp_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'picker-prefix'
      ]
      ],[[
        4,
        'width',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.col.prefixWidth;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.col.prefix;
    ck(v,1,0,currVal_1);
  });
}
function View_PickerColumnCmp_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'class',
        'picker-opt'
      ]
      ,
      [
        'disable-activated',
        ''
      ]

    ]
      ,[[
        2,
        'picker-opt-disabled',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.optClick($event,v.context.index)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit.disabled;
    ck(v,0,0,currVal_0);
    const currVal_1:any = v.context.$implicit.text;
    ck(v,1,0,currVal_1);
  });
}
function View_PickerColumnCmp_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'picker-suffix'
      ]
      ],[[
        4,
        'width',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.col.suffixWidth;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.col.suffix;
    ck(v,1,0,currVal_1);
  });
}
export function View_PickerColumnCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{colEle: 0}),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerColumnCmp_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'colEle',
        1
      ]

    ]
      ,(null as any),2,'div',[[
        'class',
        'picker-opts'
      ]
      ],[[
        4,
        'max-width',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerColumnCmp_2)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerColumnCmp_3)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import2.PickerColumnCmp = v.component;
    const currVal_0:any = co.col.prefix;
    ck(v,2,0,currVal_0);
    const currVal_2:any = co.col.options;
    ck(v,5,0,currVal_2);
    const currVal_3:any = co.col.suffix;
    ck(v,7,0,currVal_3);
  },(ck,v) => {
    var co:import2.PickerColumnCmp = v.component;
    const currVal_1:any = co.col.optionsWidth;
    ck(v,3,0,currVal_1);
  });
}
function View_PickerColumnCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'picker-col'
      ]
    ],[
      [
        4,
        'max-width',
        (null as any)
      ]
      ,
      [
        2,
        'picker-opts-left',
        (null as any)
      ]
      ,
      [
        2,
        'picker-opts-right',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_PickerColumnCmp_0,RenderType_PickerColumnCmp)),
    import0.ɵdid(4374528,(null as any),0,import2.PickerColumnCmp,[
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.NgZone,
      import5.Haptic,
      import4.Platform,
      import6.DomController
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).col.columnWidth;
    const currVal_1:any = (import0.ɵnov(v,1).col.align == 'left');
    const currVal_2:any = (import0.ɵnov(v,1).col.align == 'right');
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const PickerColumnCmpNgFactory:import0.ComponentFactory<import2.PickerColumnCmp> = import0.ɵccf('.picker-col',import2.PickerColumnCmp,View_PickerColumnCmp_Host_0,{col: 'col'},{ionChange: 'ionChange'},([] as any[]));
