// ionic-angular/components/picker/picker-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '../button/button.ngfactory';
import * as import3 from 'ionic-angular/components/button/button';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from './picker-column.ngfactory';
import * as import6 from 'ionic-angular/components/picker/picker-column';
import * as import7 from 'ionic-angular/platform/platform';
import * as import8 from 'ionic-angular/tap-click/haptic';
import * as import9 from 'ionic-angular/platform/dom-controller';
import * as import10 from 'ionic-angular/components/picker/picker-component';
import * as import11 from 'ionic-angular/components/backdrop/backdrop';
import * as import12 from 'ionic-angular/navigation/view-controller';
import * as import13 from 'ionic-angular/gestures/gesture-controller';
import * as import14 from 'ionic-angular/navigation/nav-params';
const styles_PickerCmp:any[] = ([] as any[]);
export const RenderType_PickerCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PickerCmp,
  data: {}
}
);
function View_PickerCmp_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'picker-toolbar-button'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'button',[
      [
        'class',
        'picker-button'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'ion-button',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.btnClick(v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import2.View_Button_0,import2.RenderType_Button)),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(1097728,(null as any),0,import3.Button,[
      [
        8,
        ''
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{clear: [
        0,
        'clear'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,[
      '\n            ',
      '\n          '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = 'picker-toolbar-button';
    const currVal_1:any = v.context.$implicit.cssRole;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = 'picker-button';
    const currVal_3:any = v.context.$implicit.cssClass;
    ck(v,4,0,currVal_2,currVal_3);
    const currVal_4:any = '';
    ck(v,5,0,currVal_4);
  },(ck,v) => {
    const currVal_5:any = v.context.$implicit.text;
    ck(v,6,0,currVal_5);
  });
}
function View_PickerCmp_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'picker-col'
      ]
    ],[
      [
        4,
        'max-width',
        (null as any)
      ]
      ,
      [
        2,
        'picker-opts-left',
        (null as any)
      ]
      ,
      [
        2,
        'picker-opts-right',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'ionChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('ionChange' === en)) {
        const pd_0:any = ((<any>co._colChange($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import5.View_PickerColumnCmp_0,import5.RenderType_PickerColumnCmp)),
      import0.ɵdid(4374528,[[
        1,
        4
      ]
    ],0,import6.PickerColumnCmp,[
      import4.Config,
      import7.Platform,
      import0.ElementRef,
      import0.NgZone,
      import8.Haptic,
      import7.Platform,
      import9.DomController
    ]
      ,{col: [
        0,
        'col'
      ]
    },{ionChange: 'ionChange'})
  ]
  ,(ck,v) => {
    const currVal_3:any = v.context.$implicit;
    ck(v,1,0,currVal_3);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).col.columnWidth;
    const currVal_1:any = (import0.ɵnov(v,1).col.align == 'left');
    const currVal_2:any = (import0.ɵnov(v,1).col.align == 'right');
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_PickerCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(671088640,1,{_cols: 1}),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'role',
        'presentation'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import10.PickerCmp = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.bdClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import11.Backdrop,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),17,'div',[[
        'class',
        'picker-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'picker-toolbar'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerCmp_1)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'picker-columns'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'picker-above-highlight'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_PickerCmp_2)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'picker-below-highlight'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import10.PickerCmp = v.component;
    const currVal_0:any = co.d.buttons;
    ck(v,10,0,currVal_0);
    const currVal_1:any = co.d.columns;
    ck(v,18,0,currVal_1);
  },(null as any));
}
function View_PickerCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-picker-cmp',[[
        'role',
        'dialog'
      ]
      ],(null as any),[[
        'body',
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('body:keyup' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._keyUp($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_PickerCmp_0,RenderType_PickerCmp)),
    import0.ɵdid(180224,(null as any),0,import10.PickerCmp,[
      import12.ViewController,
      import0.ElementRef,
      import4.Config,
      import13.GestureController,
      import14.NavParams,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PickerCmpNgFactory:import0.ComponentFactory<import10.PickerCmp> = import0.ɵccf('ion-picker-cmp',import10.PickerCmp,View_PickerCmp_Host_0,{},{},([] as any[]));
