// ionic-angular/components/button/button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/button/button';
import * as import2 from 'ionic-angular/config/config';
const styles_Button:any[] = ([] as any[]);
export const RenderType_Button:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Button,
  data: {}
}
);
export function View_Button_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'button-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'button-effect'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_Button_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'ion-button',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_Button_0,RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import1.Button,[
      [
        8,
        ''
      ]
      ,
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ButtonNgFactory:import0.ComponentFactory<import1.Button> = import0.ɵccf('[ion-button]',import1.Button,View_Button_Host_0,{
  color: 'color',
  mode: 'mode',
  large: 'large',
  small: 'small',
  default: 'default',
  outline: 'outline',
  clear: 'clear',
  solid: 'solid',
  round: 'round',
  block: 'block',
  full: 'full',
  strong: 'strong'
}
,{},['*']);
