// ionic-angular/components/item/item-sliding.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/item/item-sliding';
import * as import2 from 'ionic-angular/components/list/list';
import * as import3 from 'ionic-angular/platform/platform';
const styles_ItemSliding:any[] = ([] as any[]);
export const RenderType_ItemSliding:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ItemSliding,
  data: {}
}
);
export function View_ItemSliding_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    '])),
    import0.ɵncd((null as any),1),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(null as any),(null as any));
}
function View_ItemSliding_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'ion-item-sliding',([] as any[]),(null as any),(null as any),(null as any),View_ItemSliding_0,RenderType_ItemSliding)),
    import0.ɵdid(49152,(null as any),2,import1.ItemSliding,[
      [
        2,
        import2.List
      ]
      ,
      import3.Platform,
      import0.Renderer,
      import0.ElementRef,
      import0.NgZone
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(335544320,1,{item: 0}),
    import0.ɵqud(603979776,2,{_itemOptions: 1})
  ]
  ,(null as any),(null as any));
}
export const ItemSlidingNgFactory:import0.ComponentFactory<import1.ItemSliding> = import0.ɵccf('ion-item-sliding',import1.ItemSliding,View_ItemSliding_Host_0,{},{ionDrag: 'ionDrag'},[
  'ion-item,[ion-item]',
  'ion-item-options'
]
);
