// ionic-angular/components/item/reorder.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/icon/icon';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from 'ionic-angular/components/item/reorder';
const styles_Reorder:any[] = ([] as any[]);
export const RenderType_Reorder:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Reorder,
  data: {}
}
);
export function View_Reorder_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[
      [
        'name',
        'reorder'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_1:any = 'reorder';
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    ck(v,0,0,currVal_0);
  });
}
function View_Reorder_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-reorder',([] as any[]),(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_Reorder_0,RenderType_Reorder)),
    import0.ɵdid(49152,(null as any),0,import3.Reorder,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ReorderNgFactory:import0.ComponentFactory<import3.Reorder> = import0.ɵccf('ion-reorder',import3.Reorder,View_Reorder_Host_0,{},{},([] as any[]));
