// ionic-angular/components/toolbar/navbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/toolbar/navbar';
import * as import3 from '../button/button.ngfactory';
import * as import4 from 'ionic-angular/components/button/button';
import * as import5 from 'ionic-angular/config/config';
import * as import6 from 'ionic-angular/components/icon/icon';
import * as import7 from 'ionic-angular/components/app/app';
import * as import8 from 'ionic-angular/navigation/view-controller';
import * as import9 from 'ionic-angular/navigation/nav-controller';
const styles_Navbar:any[] = ([] as any[]);
export const RenderType_Navbar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Navbar,
  data: {}
}
);
export function View_Navbar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'toolbar-background'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),8,'button',[
      [
        'class',
        'back-button'
      ]
      ,
      [
        'ion-button',
        'bar-button'
      ]

    ]
      ,[[
        8,
        'hidden',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.Navbar = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.backButtonClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import3.View_Button_0,import3.RenderType_Button)),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(1097728,(null as any),0,import4.Button,[
      [
        8,
        'bar-button'
      ]
      ,
      import5.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵeld(0,(null as any),0,2,'ion-icon',[
      [
        'class',
        'back-button-icon'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(147456,(null as any),0,import6.Icon,[
      import5.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),0,2,'span',[[
        'class',
        'back-button-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    import0.ɵncd((null as any),0),
    import0.ɵncd((null as any),1),
    import0.ɵncd((null as any),2),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'toolbar-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵncd((null as any),3)
  ]
  ,(ck,v) => {
    var co:import2.Navbar = v.component;
    const currVal_0:any = 'toolbar-background';
    const currVal_1:any = ('toolbar-background-' + co._mode);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_3:any = 'back-button';
    const currVal_4:any = ('back-button-' + co._mode);
    ck(v,3,0,currVal_3,currVal_4);
    const currVal_6:any = 'back-button-icon';
    const currVal_7:any = ('back-button-icon-' + co._mode);
    ck(v,6,0,currVal_6,currVal_7);
    const currVal_8:any = co._bbIcon;
    ck(v,7,0,currVal_8);
    const currVal_9:any = 'back-button-text';
    const currVal_10:any = ('back-button-text-' + co._mode);
    ck(v,9,0,currVal_9,currVal_10);
    const currVal_12:any = 'toolbar-content';
    const currVal_13:any = ('toolbar-content-' + co._mode);
    ck(v,15,0,currVal_12,currVal_13);
  },(ck,v) => {
    var co:import2.Navbar = v.component;
    const currVal_2:any = co._hideBb;
    ck(v,2,0,currVal_2);
    const currVal_5:any = import0.ɵnov(v,7)._hidden;
    ck(v,5,0,currVal_5);
    const currVal_11:any = co._backText;
    ck(v,10,0,currVal_11);
  });
}
function View_Navbar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-navbar',[[
        'class',
        'toolbar'
      ]
    ],[
      [
        8,
        'hidden',
        0
      ]
      ,
      [
        2,
        'statusbar-padding',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Navbar_0,RenderType_Navbar)),
    import0.ɵdid(49152,(null as any),0,import2.Navbar,[
      import7.App,
      [
        2,
        import8.ViewController
      ]
      ,
      [
        2,
        import9.NavController
      ]
      ,
      import5.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    const currVal_1:any = import0.ɵnov(v,1)._sbPadding;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const NavbarNgFactory:import0.ComponentFactory<import2.Navbar> = import0.ɵccf('ion-navbar',import2.Navbar,View_Navbar_Host_0,{
  color: 'color',
  mode: 'mode',
  hideBackButton: 'hideBackButton'
}
,{},[
  '[menuToggle],ion-buttons[left]',
  'ion-buttons[start]',
  'ion-buttons[end],ion-buttons[right]',
  '*'
]
);
