// ionic-angular/components/toolbar/toolbar-title.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/toolbar/toolbar-title';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/components/toolbar/toolbar';
import * as import5 from 'ionic-angular/components/toolbar/navbar';
const styles_ToolbarTitle:any[] = ([] as any[]);
export const RenderType_ToolbarTitle:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ToolbarTitle,
  data: {}
}
);
export function View_ToolbarTitle_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'toolbar-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵncd((null as any),0)
  ]
  ,(ck,v) => {
    var co:import2.ToolbarTitle = v.component;
    const currVal_0:any = 'toolbar-title';
    const currVal_1:any = ('toolbar-title-' + co._mode);
    ck(v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_ToolbarTitle_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-title',([] as any[]),(null as any),(null as any),(null as any),View_ToolbarTitle_0,RenderType_ToolbarTitle)),
    import0.ɵdid(49152,(null as any),0,import2.ToolbarTitle,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import4.Toolbar
      ]
      ,
      [
        2,
        import5.Navbar
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ToolbarTitleNgFactory:import0.ComponentFactory<import2.ToolbarTitle> = import0.ɵccf('ion-title',import2.ToolbarTitle,View_ToolbarTitle_Host_0,{
  color: 'color',
  mode: 'mode'
}
,{},['*']);
