// ionic-angular/components/toolbar/toolbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/toolbar/toolbar';
import * as import3 from 'ionic-angular/config/config';
const styles_Toolbar:any[] = ([] as any[]);
export const RenderType_Toolbar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Toolbar,
  data: {}
}
);
export function View_Toolbar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'toolbar-background'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵncd((null as any),0),
    import0.ɵncd((null as any),1),
    import0.ɵncd((null as any),2),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'toolbar-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵncd((null as any),3)
  ]
  ,(ck,v) => {
    var co:import2.Toolbar = v.component;
    const currVal_0:any = 'toolbar-background';
    const currVal_1:any = ('toolbar-background-' + co._mode);
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = 'toolbar-content';
    const currVal_3:any = ('toolbar-content-' + co._mode);
    ck(v,6,0,currVal_2,currVal_3);
  },(null as any));
}
function View_Toolbar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-toolbar',[[
        'class',
        'toolbar'
      ]
      ],[[
        2,
        'statusbar-padding',
        (null as any)
      ]
    ],(null as any),(null as any),View_Toolbar_0,RenderType_Toolbar)),
    import0.ɵdid(49152,(null as any),0,import2.Toolbar,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._sbPadding;
    ck(v,0,0,currVal_0);
  });
}
export const ToolbarNgFactory:import0.ComponentFactory<import2.Toolbar> = import0.ɵccf('ion-toolbar',import2.Toolbar,View_Toolbar_Host_0,{
  color: 'color',
  mode: 'mode'
}
,{},[
  '[menuToggle],ion-buttons[left]',
  'ion-buttons[start]',
  'ion-buttons[end],ion-buttons[right]',
  '*'
]
);
