// ionic-angular/module.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/module';
import * as import2 from '@angular/common';
import * as import3 from '@angular/forms';
import * as import4 from './components/action-sheet/action-sheet-component.ngfactory';
import * as import5 from './components/alert/alert-component.ngfactory';
import * as import6 from './components/app/app-root.ngfactory';
import * as import7 from './components/loading/loading-component.ngfactory';
import * as import8 from './components/modal/modal-component.ngfactory';
import * as import9 from './components/picker/picker-component.ngfactory';
import * as import10 from './components/popover/popover-component.ngfactory';
import * as import11 from './components/select/select-popover-component.ngfactory';
import * as import12 from './components/toast/toast-component.ngfactory';
class IonicModuleInjector extends import0.ɵNgModuleInjector<import1.IonicModule> {
  _CommonModule_0:import2.CommonModule;
  _ɵba_1:import3.ɵba;
  _FormsModule_2:import3.FormsModule;
  _ReactiveFormsModule_3:import3.ReactiveFormsModule;
  _IonicModule_4:import1.IonicModule;
  __NgLocalization_5:import2.NgLocaleLocalization;
  __ɵi_6:import3.ɵi;
  __FormBuilder_7:import3.FormBuilder;
  constructor(parent:import0.Injector) {
    super(parent,[
      import4.ActionSheetCmpNgFactory,
      import5.AlertCmpNgFactory,
      import6.IonicAppNgFactory,
      import7.LoadingCmpNgFactory,
      import8.ModalCmpNgFactory,
      import9.PickerCmpNgFactory,
      import10.PopoverCmpNgFactory,
      import11.SelectPopoverNgFactory,
      import12.ToastCmpNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_5():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_5 == null)) { (this.__NgLocalization_5 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_5;
  }
  get _ɵi_6():import3.ɵi {
    if ((this.__ɵi_6 == null)) { (this.__ɵi_6 = new import3.ɵi()); }
    return this.__ɵi_6;
  }
  get _FormBuilder_7():import3.FormBuilder {
    if ((this.__FormBuilder_7 == null)) { (this.__FormBuilder_7 = new import3.FormBuilder()); }
    return this.__FormBuilder_7;
  }
  createInternal():import1.IonicModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ɵba_1 = new import3.ɵba();
    this._FormsModule_2 = new import3.FormsModule();
    this._ReactiveFormsModule_3 = new import3.ReactiveFormsModule();
    this._IonicModule_4 = new import1.IonicModule();
    return this._IonicModule_4;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ɵba)) { return this._ɵba_1; }
    if ((token === import3.FormsModule)) { return this._FormsModule_2; }
    if ((token === import3.ReactiveFormsModule)) { return this._ReactiveFormsModule_3; }
    if ((token === import1.IonicModule)) { return this._IonicModule_4; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_5; }
    if ((token === import3.ɵi)) { return this._ɵi_6; }
    if ((token === import3.FormBuilder)) { return this._FormBuilder_7; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const IonicModuleNgFactory:import0.NgModuleFactory<import1.IonicModule> = new import0.NgModuleFactory<any>(IonicModuleInjector,import1.IonicModule);
class IonicPageModuleInjector extends import0.ɵNgModuleInjector<import1.IonicPageModule> {
  _CommonModule_0:import2.CommonModule;
  _ɵba_1:import3.ɵba;
  _FormsModule_2:import3.FormsModule;
  _ReactiveFormsModule_3:import3.ReactiveFormsModule;
  _IonicModule_4:import1.IonicModule;
  _IonicPageModule_5:import1.IonicPageModule;
  __NgLocalization_6:import2.NgLocaleLocalization;
  __ɵi_7:import3.ɵi;
  __FormBuilder_8:import3.FormBuilder;
  constructor(parent:import0.Injector) {
    super(parent,[
      import4.ActionSheetCmpNgFactory,
      import5.AlertCmpNgFactory,
      import6.IonicAppNgFactory,
      import7.LoadingCmpNgFactory,
      import8.ModalCmpNgFactory,
      import9.PickerCmpNgFactory,
      import10.PopoverCmpNgFactory,
      import11.SelectPopoverNgFactory,
      import12.ToastCmpNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_6():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_6 == null)) { (this.__NgLocalization_6 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_6;
  }
  get _ɵi_7():import3.ɵi {
    if ((this.__ɵi_7 == null)) { (this.__ɵi_7 = new import3.ɵi()); }
    return this.__ɵi_7;
  }
  get _FormBuilder_8():import3.FormBuilder {
    if ((this.__FormBuilder_8 == null)) { (this.__FormBuilder_8 = new import3.FormBuilder()); }
    return this.__FormBuilder_8;
  }
  createInternal():import1.IonicPageModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ɵba_1 = new import3.ɵba();
    this._FormsModule_2 = new import3.FormsModule();
    this._ReactiveFormsModule_3 = new import3.ReactiveFormsModule();
    this._IonicModule_4 = new import1.IonicModule();
    this._IonicPageModule_5 = new import1.IonicPageModule();
    return this._IonicPageModule_5;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ɵba)) { return this._ɵba_1; }
    if ((token === import3.FormsModule)) { return this._FormsModule_2; }
    if ((token === import3.ReactiveFormsModule)) { return this._ReactiveFormsModule_3; }
    if ((token === import1.IonicModule)) { return this._IonicModule_4; }
    if ((token === import1.IonicPageModule)) { return this._IonicPageModule_5; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_6; }
    if ((token === import3.ɵi)) { return this._ɵi_7; }
    if ((token === import3.FormBuilder)) { return this._FormBuilder_8; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const IonicPageModuleNgFactory:import0.NgModuleFactory<import1.IonicPageModule> = new import0.NgModuleFactory<any>(IonicPageModuleInjector,import1.IonicPageModule);
