// ionic-angular/components/spinner/spinner.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/spinner/spinner';
import * as import3 from 'ionic-angular/config/config';
const styles_Spinner:any[] = ([] as any[]);
export const RenderType_Spinner:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Spinner,
  data: {}
}
);
function View_Spinner_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,':svg:svg',[[
        'viewBox',
        '0 0 64 64'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,':svg:circle',[[
        'transform',
        'translate(32,32)'
      ]
      ],[[
        1,
        'r',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.style;
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    const currVal_1:any = v.context.$implicit.r;
    ck(v,2,0,currVal_1);
  });
}
function View_Spinner_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,':svg:svg',[[
        'viewBox',
        '0 0 64 64'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,':svg:line',[[
        'transform',
        'translate(32,32)'
      ]
    ],[
      [
        1,
        'y1',
        0
      ]
      ,
      [
        1,
        'y2',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    const currVal_0:any = v.context.$implicit.style;
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    const currVal_1:any = v.context.$implicit.y1;
    const currVal_2:any = v.context.$implicit.y2;
    ck(v,2,0,currVal_1,currVal_2);
  });
}
export function View_Spinner_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Spinner_1)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Spinner_2)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import2.Spinner = v.component;
    const currVal_0:any = co._c;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co._l;
    ck(v,3,0,currVal_1);
  },(null as any));
}
function View_Spinner_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',([] as any[]),[[
        2,
        'spinner-paused',
        (null as any)
      ]
    ],(null as any),(null as any),View_Spinner_0,RenderType_Spinner)),
    import0.ɵdid(114688,(null as any),0,import2.Spinner,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._paused;
    ck(v,0,0,currVal_0);
  });
}
export const SpinnerNgFactory:import0.ComponentFactory<import2.Spinner> = import0.ɵccf('ion-spinner',import2.Spinner,View_Spinner_Host_0,{
  color: 'color',
  mode: 'mode',
  name: 'name',
  duration: 'duration',
  paused: 'paused'
}
,{},([] as any[]));
