// ionic-angular/components/content/content.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/content/content';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from 'ionic-angular/platform/platform';
import * as import4 from 'ionic-angular/platform/dom-controller';
import * as import5 from 'ionic-angular/components/app/app';
import * as import6 from 'ionic-angular/platform/keyboard';
import * as import7 from 'ionic-angular/navigation/view-controller';
import * as import8 from 'ionic-angular/navigation/nav-controller';
const styles_Content:any[] = ([] as any[]);
export const RenderType_Content:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Content,
  data: {}
}
);
export function View_Content_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(402653184,1,{_fixedContent: 0}),
    import0.ɵqud(402653184,2,{_scrollContent: 0}),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'fixedContent',
        1
      ]

    ]
      ,(null as any),1,'div',[[
        'class',
        'fixed-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'scrollContent',
        1
      ]

    ]
      ,(null as any),1,'div',[[
        'class',
        'scroll-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),1),
    import0.ɵncd((null as any),2)
  ]
  ,(null as any),(null as any));
}
function View_Content_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-content',([] as any[]),[
      [
        2,
        'statusbar-padding',
        (null as any)
      ]
      ,
      [
        2,
        'has-refresher',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Content_0,RenderType_Content)),
    import0.ɵdid(4374528,(null as any),0,import1.Content,[
      import2.Config,
      import3.Platform,
      import4.DomController,
      import0.ElementRef,
      import0.Renderer,
      import5.App,
      import6.Keyboard,
      import0.NgZone,
      [
        2,
        import7.ViewController
      ]
      ,
      [
        2,
        import8.NavController
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).statusbarPadding;
    const currVal_1:any = import0.ɵnov(v,1)._hasRefresher;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const ContentNgFactory:import0.ComponentFactory<import1.Content> = import0.ɵccf('ion-content',import1.Content,View_Content_Host_0,{
  color: 'color',
  mode: 'mode',
  fullscreen: 'fullscreen',
  scrollDownOnLoad: 'scrollDownOnLoad'
}
,{
  ionScrollStart: 'ionScrollStart',
  ionScroll: 'ionScroll',
  ionScrollEnd: 'ionScrollEnd'
}
,[
  '[ion-fixed],ion-fab',
  '*',
  'ion-refresher'
]
);
