// ionic-angular/components/img/img.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/img/img';
import * as import2 from 'ionic-angular/platform/platform';
import * as import3 from 'ionic-angular/components/content/content';
import * as import4 from 'ionic-angular/platform/dom-controller';
const styles_Img:any[] = ([] as any[]);
export const RenderType_Img:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Img,
  data: {}
}
);
export function View_Img_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'img',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_Img_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-img',([] as any[]),(null as any),(null as any),(null as any),View_Img_0,RenderType_Img)),
    import0.ɵdid(1228800,(null as any),0,import1.Img,[
      import0.ElementRef,
      import0.Renderer,
      import2.Platform,
      [
        2,
        import3.Content
      ]
      ,
      import4.DomController
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const ImgNgFactory:import0.ComponentFactory<import1.Img> = import0.ɵccf('ion-img',import1.Img,View_Img_Host_0,{
  src: 'src',
  bounds: 'bounds',
  cache: 'cache',
  width: 'width',
  height: 'height',
  alt: 'alt'
}
,{},([] as any[]));
