// ionic-angular/components/slides/slides.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/slides/slides';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from 'ionic-angular/platform/platform';
import * as import4 from 'ionic-angular/navigation/view-controller';
const styles_Slides:any[] = ([] as any[]);
export const RenderType_Slides:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Slides,
  data: {}
}
);
export function View_Slides_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'swiper-container'
      ]
      ],[[
        1,
        'dir',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'swiper-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'swiper-pagination'
      ]
      ],[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.Slides = v.component;
    const currVal_0:any = (co._rtl? 'rtl': (null as any));
    ck(v,0,0,currVal_0);
    const currVal_1:boolean = !co.pager;
    ck(v,3,0,currVal_1);
  });
}
function View_Slides_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-slides',([] as any[]),(null as any),(null as any),(null as any),View_Slides_0,RenderType_Slides)),
    import0.ɵdid(1228800,(null as any),0,import1.Slides,[
      import2.Config,
      import3.Platform,
      import0.NgZone,
      [
        2,
        import4.ViewController
      ]
      ,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const SlidesNgFactory:import0.ComponentFactory<import1.Slides> = import0.ɵccf('ion-slides',import1.Slides,View_Slides_Host_0,{
  color: 'color',
  mode: 'mode',
  autoplay: 'autoplay',
  control: 'control',
  effect: 'effect',
  direction: 'direction',
  initialSlide: 'initialSlide',
  loop: 'loop',
  pager: 'pager',
  dir: 'dir',
  paginationType: 'paginationType',
  parallax: 'parallax',
  speed: 'speed',
  zoom: 'zoom',
  spaceBetween: 'spaceBetween',
  slidesPerView: 'slidesPerView',
  centeredSlides: 'centeredSlides'
}
,{
  ionSlideWillChange: 'ionSlideWillChange',
  ionSlideDidChange: 'ionSlideDidChange',
  ionSlideDrag: 'ionSlideDrag',
  ionSlideReachStart: 'ionSlideReachStart',
  ionSlideReachEnd: 'ionSlideReachEnd',
  ionSlideAutoplay: 'ionSlideAutoplay',
  ionSlideAutoplayStart: 'ionSlideAutoplayStart',
  ionSlideAutoplayStop: 'ionSlideAutoplayStop',
  ionSlideNextStart: 'ionSlideNextStart',
  ionSlidePrevStart: 'ionSlidePrevStart',
  ionSlideNextEnd: 'ionSlideNextEnd',
  ionSlidePrevEnd: 'ionSlidePrevEnd',
  ionSlideTap: 'ionSlideTap',
  ionSlideDoubleTap: 'ionSlideDoubleTap'
}
,['*']);
