// ionic-angular/components/slides/slide.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/slides/slide';
import * as import2 from 'ionic-angular/components/slides/slides';
const styles_Slide:any[] = ([] as any[]);
export const RenderType_Slide:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Slide,
  data: {}
}
);
export function View_Slide_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'slide-zoom'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0)
  ]
  ,(null as any),(null as any));
}
function View_Slide_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),View_Slide_0,RenderType_Slide)),
    import0.ɵdid(180224,(null as any),0,import1.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import2.Slides
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const SlideNgFactory:import0.ComponentFactory<import1.Slide> = import0.ɵccf('ion-slide',import1.Slide,View_Slide_Host_0,{},{},['*']);
