// ionic-angular/components/action-sheet/action-sheet-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/icon/icon';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from '../button/button.ngfactory';
import * as import4 from '@angular/common';
import * as import5 from 'ionic-angular/components/button/button';
import * as import6 from 'ionic-angular/components/action-sheet/action-sheet-component';
import * as import7 from 'ionic-angular/components/backdrop/backdrop';
import * as import8 from 'ionic-angular/navigation/view-controller';
import * as import9 from 'ionic-angular/gestures/gesture-controller';
import * as import10 from 'ionic-angular/navigation/nav-params';
const styles_ActionSheetCmp:any[] = ([] as any[]);
export const RenderType_ActionSheetCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ActionSheetCmp,
  data: {}
}
);
function View_ActionSheetCmp_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'action-sheet-title'
      ]
      ],[[
        8,
        'id',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.hdrId,'');
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.d.title;
    ck(v,1,0,currVal_1);
  });
}
function View_ActionSheetCmp_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'action-sheet-sub-title'
      ]
      ],[[
        8,
        'id',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.descId,'');
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.d.subTitle;
    ck(v,1,0,currVal_1);
  });
}
function View_ActionSheetCmp_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[
      [
        'class',
        'action-sheet-icon'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_1:any = (<any>v.parent).context.$implicit.icon;
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    ck(v,0,0,currVal_0);
  });
}
function View_ActionSheetCmp_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'button',[
      [
        'class',
        'disable-hover'
      ]
      ,
      [
        'ion-button',
        'action-sheet-button'
      ]

    ]
      ,[[
        1,
        'icon-start',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.click(v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import3.View_Button_0,import3.RenderType_Button)),
    import0.ɵdid(278528,(null as any),0,import4.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(1097728,(null as any),0,import5.Button,[
      [
        8,
        'action-sheet-button'
      ]
      ,
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),0,1,(null as any),View_ActionSheetCmp_4)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    const currVal_1:any = 'disable-hover';
    const currVal_2:any = v.context.$implicit.cssClass;
    ck(v,1,0,currVal_1,currVal_2);
    const currVal_3:any = v.context.$implicit.icon;
    ck(v,4,0,currVal_3);
  },(ck,v) => {
    const currVal_0:any = (v.context.$implicit.icon? '': (null as any));
    ck(v,0,0,currVal_0);
    const currVal_4:any = v.context.$implicit.text;
    ck(v,5,0,currVal_4);
  });
}
function View_ActionSheetCmp_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[
      [
        'class',
        'action-sheet-icon'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.cancelButton.icon;
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    ck(v,0,0,currVal_0);
  });
}
function View_ActionSheetCmp_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'action-sheet-group'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'button',[
      [
        'class',
        'action-sheet-cancel disable-hover'
      ]
      ,
      [
        'ion-button',
        'action-sheet-button'
      ]

    ]
      ,[[
        1,
        'icon-start',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.click(co.cancelButton)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import3.View_Button_0,import3.RenderType_Button)),
    import0.ɵdid(278528,(null as any),0,import4.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵdid(1097728,(null as any),0,import5.Button,[
      [
        8,
        'action-sheet-button'
      ]
      ,
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),0,1,(null as any),View_ActionSheetCmp_6)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'action-sheet-cancel disable-hover';
    const currVal_2:any = co.cancelButton.cssClass;
    ck(v,2,0,currVal_1,currVal_2);
    const currVal_3:any = co.cancelButton.icon;
    ck(v,5,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (co.cancelButton.icon? '': (null as any));
    ck(v,1,0,currVal_0);
    const currVal_4:any = co.cancelButton.text;
    ck(v,6,0,currVal_4);
  });
}
export function View_ActionSheetCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'role',
        'presentation'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,[[
        2,
        'backdrop-no-tappable',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import6.ActionSheetCmp = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.bdClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import7.Backdrop,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'action-sheet-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'action-sheet-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'class',
        'action-sheet-group'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ActionSheetCmp_1)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ActionSheetCmp_2)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ActionSheetCmp_3)),
    import0.ɵdid(802816,(null as any),0,import4.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ActionSheetCmp_5)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import6.ActionSheetCmp = v.component;
    const currVal_1:any = co.d.title;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.d.subTitle;
    ck(v,8,0,currVal_2);
    const currVal_3:any = co.d.buttons;
    ck(v,10,0,currVal_3);
    const currVal_4:any = co.cancelButton;
    ck(v,12,0,currVal_4);
  },(ck,v) => {
    var co:import6.ActionSheetCmp = v.component;
    const currVal_0:boolean = !co.d.enableBackdropDismiss;
    ck(v,0,0,currVal_0);
  });
}
function View_ActionSheetCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-action-sheet',[[
        'role',
        'dialog'
      ]
    ],[
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'aria-describedby',
        0
      ]

    ]
      ,[[
        'body',
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('body:keyup' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).keyUp($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_ActionSheetCmp_0,RenderType_ActionSheetCmp)),
    import0.ɵdid(180224,(null as any),0,import6.ActionSheetCmp,[
      import8.ViewController,
      import2.Config,
      import0.ElementRef,
      import9.GestureController,
      import10.NavParams,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hdrId;
    const currVal_1:any = import0.ɵnov(v,1).descId;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const ActionSheetCmpNgFactory:import0.ComponentFactory<import6.ActionSheetCmp> = import0.ɵccf('ion-action-sheet',import6.ActionSheetCmp,View_ActionSheetCmp_Host_0,{},{},([] as any[]));
