// ionic-angular/components/loading/loading-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../spinner/spinner.ngfactory';
import * as import2 from 'ionic-angular/components/spinner/spinner';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/components/loading/loading-component';
import * as import5 from 'ionic-angular/components/backdrop/backdrop';
import * as import6 from '@angular/common';
import * as import7 from 'ionic-angular/navigation/view-controller';
import * as import8 from 'ionic-angular/gestures/gesture-controller';
import * as import9 from 'ionic-angular/navigation/nav-params';
const styles_LoadingCmp:any[] = ([] as any[]);
export const RenderType_LoadingCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_LoadingCmp,
  data: {}
}
);
function View_LoadingCmp_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'loading-spinner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',([] as any[]),[[
        2,
        'spinner-paused',
        (null as any)
      ]
    ],(null as any),(null as any),import1.View_Spinner_0,import1.RenderType_Spinner)),
    import0.ɵdid(114688,(null as any),0,import2.Spinner,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.d.spinner;
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2)._paused;
    ck(v,1,0,currVal_0);
  });
}
function View_LoadingCmp_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'loading-content'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.d.content;
    ck(v,0,0,currVal_0);
  });
}
export function View_LoadingCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'role',
        'presentation'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
    ,[
      [
        8,
        'hidden',
        0
      ]
      ,
      [
        2,
        'backdrop-no-tappable',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import4.LoadingCmp = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.bdClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import5.Backdrop,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'loading-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_LoadingCmp_1)),
    import0.ɵdid(16384,(null as any),0,import6.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_LoadingCmp_2)),
    import0.ɵdid(16384,(null as any),0,import6.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import4.LoadingCmp = v.component;
    const currVal_2:any = co.showSpinner;
    ck(v,4,0,currVal_2);
    const currVal_3:any = co.d.content;
    ck(v,6,0,currVal_3);
  },(ck,v) => {
    var co:import4.LoadingCmp = v.component;
    const currVal_0:boolean = !co.d.showBackdrop;
    const currVal_1:boolean = !co.d.enableBackdropDismiss;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_LoadingCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-loading',[[
        'role',
        'dialog'
      ]
      ],(null as any),[[
        'body',
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('body:keyup' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).keyUp($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_LoadingCmp_0,RenderType_LoadingCmp)),
    import0.ɵdid(245760,(null as any),0,import4.LoadingCmp,[
      import7.ViewController,
      import3.Config,
      import0.ElementRef,
      import8.GestureController,
      import9.NavParams,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const LoadingCmpNgFactory:import0.ComponentFactory<import4.LoadingCmp> = import0.ɵccf('ion-loading',import4.LoadingCmp,View_LoadingCmp_Host_0,{},{},([] as any[]));
