// ionic-angular/components/toggle/toggle.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../button/button.ngfactory';
import * as import2 from 'ionic-angular/components/button/button';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/components/toggle/toggle';
import * as import5 from 'ionic-angular/util/form';
import * as import6 from 'ionic-angular/platform/platform';
import * as import7 from 'ionic-angular/tap-click/haptic';
import * as import8 from 'ionic-angular/components/item/item';
import * as import9 from 'ionic-angular/gestures/gesture-controller';
import * as import10 from 'ionic-angular/platform/dom-controller';
import * as import11 from '@angular/forms';
const styles_Toggle:any[] = ([] as any[]);
export const RenderType_Toggle:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Toggle,
  data: {}
}
);
export function View_Toggle_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'toggle-icon'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'toggle-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'class',
        'item-cover'
      ]
      ,
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'ion-button',
        'item-cover'
      ]
      ,
      [
        'role',
        'checkbox'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-checked',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
    ,(null as any),(null as any),import1.View_Button_0,import1.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import2.Button,[
      [
        8,
        'item-cover'
      ]
      ,
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    var co:import4.Toggle = v.component;
    const currVal_0:any = co.id;
    const currVal_1:any = co._value;
    const currVal_2:any = co._labelId;
    const currVal_3:any = co._disabled;
    ck(v,2,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_Toggle_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-toggle',([] as any[]),[
      [
        2,
        'toggle-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'toggle-checked',
        (null as any)
      ]
      ,
      [
        2,
        'toggle-activated',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('keyup' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._keyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_Toggle_0,RenderType_Toggle)),
    import0.ɵdid(1228800,(null as any),0,import4.Toggle,[
      import5.Form,
      import3.Config,
      import6.Platform,
      import0.ElementRef,
      import0.Renderer,
      import7.Haptic,
      [
        2,
        import8.Item
      ]
      ,
      import9.GestureController,
      import10.DomController,
      import0.NgZone
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(5120,(null as any),import11.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import4.Toggle])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._disabled;
    const currVal_1:any = import0.ɵnov(v,1)._value;
    const currVal_2:any = import0.ɵnov(v,1)._activated;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const ToggleNgFactory:import0.ComponentFactory<import4.Toggle> = import0.ɵccf('ion-toggle',import4.Toggle,View_Toggle_Host_0,{
  color: 'color',
  mode: 'mode',
  disabled: 'disabled',
  checked: 'checked'
}
,{
  ionFocus: 'ionFocus',
  ionChange: 'ionChange',
  ionBlur: 'ionBlur'
}
,([] as any[]));
