// ionic-angular/components/segment/segment-button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/segment/segment-button';
const styles_SegmentButton:any[] = ([] as any[]);
export const RenderType_SegmentButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_SegmentButton,
  data: {}
}
);
export function View_SegmentButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),0),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'button-effect'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_SegmentButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-segment-button',[
      [
        'class',
        'segment-button'
      ]
      ,
      [
        'role',
        'button'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
    ,[
      [
        2,
        'segment-button-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'segment-activated',
        (null as any)
      ]
      ,
      [
        1,
        'aria-pressed',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_SegmentButton_0,RenderType_SegmentButton)),
    import0.ɵdid(114688,(null as any),0,import1.SegmentButton,([] as any[]),(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._disabled;
    const currVal_1:any = import0.ɵnov(v,1).isActive;
    const currVal_2:any = import0.ɵnov(v,1).isActive;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const SegmentButtonNgFactory:import0.ComponentFactory<import1.SegmentButton> = import0.ɵccf('ion-segment-button',import1.SegmentButton,View_SegmentButton_Host_0,{
  value: 'value',
  disabled: 'disabled'
}
,{ionSelect: 'ionSelect'},['*']);
