// ionic-angular/components/input/input.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '../button/button.ngfactory';
import * as import3 from 'ionic-angular/components/button/button';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/components/input/input';
import * as import6 from 'ionic-angular/platform/platform';
import * as import7 from 'ionic-angular/util/form';
import * as import8 from 'ionic-angular/components/app/app';
import * as import9 from 'ionic-angular/components/content/content';
import * as import10 from 'ionic-angular/components/item/item';
import * as import11 from '@angular/forms';
import * as import12 from 'ionic-angular/platform/dom-controller';
const styles_TextInput:any[] = ([] as any[]);
export const RenderType_TextInput:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TextInput,
  data: {}
}
);
function View_TextInput_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'textInput',
        1
      ]

    ]
      ,(null as any),1,'input',[[
        'class',
        'text-input'
      ]
    ],[
      [
        8,
        'type',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'min',
        0
      ]
      ,
      [
        1,
        'max',
        0
      ]
      ,
      [
        1,
        'step',
        0
      ]
      ,
      [
        1,
        'autocomplete',
        0
      ]
      ,
      [
        1,
        'autocorrect',
        0
      ]
      ,
      [
        8,
        'placeholder',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.onInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('focus' === en)) {
        const pd_2:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keydown' === en)) {
        const pd_3:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_10:any = 'text-input';
    const currVal_11:any = ('text-input-' + co._mode);
    ck(v,1,0,currVal_10,currVal_11);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co._type;
    const currVal_1:any = co._labelId;
    const currVal_2:any = co.min;
    const currVal_3:any = co.max;
    const currVal_4:any = co.step;
    const currVal_5:any = co.autocomplete;
    const currVal_6:any = co.autocorrect;
    const currVal_7:any = co.placeholder;
    const currVal_8:any = co._disabled;
    const currVal_9:any = co._readonly;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
function View_TextInput_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'textInput',
        1
      ]

    ]
      ,(null as any),1,'textarea',[[
        'class',
        'text-input'
      ]
    ],[
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'autocomplete',
        0
      ]
      ,
      [
        1,
        'autocorrect',
        0
      ]
      ,
      [
        8,
        'placeholder',
        0
      ]
      ,
      [
        8,
        'disabled',
        0
      ]
      ,
      [
        8,
        'readOnly',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.onInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.onBlur($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('focus' === en)) {
        const pd_2:any = ((<any>co.onFocus($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('keydown' === en)) {
        const pd_3:any = ((<any>co.onKeydown($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(278528,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_6:any = 'text-input';
    const currVal_7:any = ('text-input-' + co._mode);
    ck(v,1,0,currVal_6,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co._labelId;
    const currVal_1:any = co.autocomplete;
    const currVal_2:any = co.autocorrect;
    const currVal_3:any = co.placeholder;
    const currVal_4:any = co._disabled;
    const currVal_5:any = co._readonly;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
  });
}
function View_TextInput_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'class',
        'text-input-clear-icon'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'ion-button',
        ''
      ]
      ,
      [
        'tabindex',
        '-1'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,(null as any),[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clearTextInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.clearTextInput($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import2.View_Button_0,import2.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import3.Button,[
      [
        8,
        ''
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{clear: [
        0,
        'clear'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = '';
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_TextInput_4(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'input-cover'
      ]
    ],(null as any),[
      [
        (null as any),
        'touchstart'
      ]
      ,
      [
        (null as any),
        'touchend'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]
      ,
      [
        (null as any),
        'mouseup'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('touchstart' === en)) {
        const pd_0:any = ((<any>co._pointerStart($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('touchend' === en)) {
        const pd_1:any = ((<any>co._pointerEnd($event)) !== false);
        ad = (pd_1 && ad);
      }
      if (('mousedown' === en)) {
        const pd_2:any = ((<any>co._pointerStart($event)) !== false);
        ad = (pd_2 && ad);
      }
      if (('mouseup' === en)) {
        const pd_3:any = ((<any>co._pointerEnd($event)) !== false);
        ad = (pd_3 && ad);
      }
      return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
export function View_TextInput_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(671088640,1,{_native: 0}),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_2)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_3)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TextInput_4)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import5.TextInput = v.component;
    const currVal_0:boolean = !co._isTextarea;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co._isTextarea;
    ck(v,4,0,currVal_1);
    const currVal_2:any = co._clearInput;
    ck(v,6,0,currVal_2);
    const currVal_3:any = co._useAssist;
    ck(v,8,0,currVal_3);
  },(null as any));
}
function View_TextInput_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-input',([] as any[]),(null as any),(null as any),(null as any),View_TextInput_0,RenderType_TextInput)),
    import0.ɵdid(5423104,(null as any),0,import5.TextInput,[
      import4.Config,
      import6.Platform,
      import7.Form,
      import8.App,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import9.Content
      ]
      ,
      [
        2,
        import10.Item
      ]
      ,
      [
        2,
        import11.NgControl
      ]
      ,
      import12.DomController
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const TextInputNgFactory:import0.ComponentFactory<import5.TextInput> = import0.ɵccf('ion-input,ion-textarea',import5.TextInput,View_TextInput_Host_0,{
  value: 'value',
  color: 'color',
  mode: 'mode',
  disabled: 'disabled',
  clearInput: 'clearInput',
  type: 'type',
  readonly: 'readonly',
  clearOnEdit: 'clearOnEdit',
  autocomplete: 'autocomplete',
  autocorrect: 'autocorrect',
  placeholder: 'placeholder',
  min: 'min',
  max: 'max',
  step: 'step'
}
,{
  ionFocus: 'ionFocus',
  ionChange: 'ionChange',
  ionBlur: 'ionBlur',
  input: 'input',
  blur: 'blur',
  focus: 'focus'
}
,([] as any[]));
