// ionic-angular/components/tabs/tab-button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/icon/icon';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from 'ionic-angular/components/badge/badge';
import * as import4 from '@angular/common';
import * as import5 from 'ionic-angular/components/tabs/tab-button';
const styles_TabButton:any[] = ([] as any[]);
export const RenderType_TabButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_TabButton,
  data: {}
}
);
function View_TabButton_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[
      [
        'class',
        'tab-button-icon'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      name: [
        0,
        'name'
      ]
      ,
      isActive: [
        1,
        'isActive'
      ]

    }
    ,(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.tab.tabIcon;
    const currVal_2:any = co.tab.isSelected;
    ck(v,1,0,currVal_1,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    ck(v,0,0,currVal_0);
  });
}
function View_TabButton_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'tab-button-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.tab.tabTitle;
    ck(v,1,0,currVal_0);
  });
}
function View_TabButton_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-badge',[[
        'class',
        'tab-badge'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import3.Badge,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{color: [
        0,
        'color'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.tab.tabBadgeStyle;
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.tab.tabBadge;
    ck(v,2,0,currVal_1);
  });
}
export function View_TabButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabButton_1)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabButton_2)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_TabButton_3)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'button-effect'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    var co:import5.TabButton = v.component;
    const currVal_0:any = co.tab.tabIcon;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.tab.tabTitle;
    ck(v,3,0,currVal_1);
    const currVal_2:any = co.tab.tabBadge;
    ck(v,5,0,currVal_2);
  },(null as any));
}
function View_TabButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'tab-button'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-controls',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]
      ,
      [
        2,
        'has-title',
        (null as any)
      ]
      ,
      [
        2,
        'has-icon',
        (null as any)
      ]
      ,
      [
        2,
        'has-title-only',
        (null as any)
      ]
      ,
      [
        2,
        'icon-only',
        (null as any)
      ]
      ,
      [
        2,
        'has-badge',
        (null as any)
      ]
      ,
      [
        2,
        'disable-hover',
        (null as any)
      ]
      ,
      [
        2,
        'tab-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'tab-hidden',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_TabButton_0,RenderType_TabButton)),
    import0.ɵdid(114688,(null as any),0,import5.TabButton,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).tab._btnId;
    const currVal_1:any = import0.ɵnov(v,1).tab._tabId;
    const currVal_2:any = import0.ɵnov(v,1).tab.isSelected;
    const currVal_3:any = import0.ɵnov(v,1).hasTitle;
    const currVal_4:any = import0.ɵnov(v,1).hasIcon;
    const currVal_5:any = import0.ɵnov(v,1).hasTitleOnly;
    const currVal_6:any = import0.ɵnov(v,1).hasIconOnly;
    const currVal_7:any = import0.ɵnov(v,1).hasBadge;
    const currVal_8:any = import0.ɵnov(v,1).disHover;
    const currVal_9:boolean = !import0.ɵnov(v,1).tab.enabled;
    const currVal_10:boolean = !import0.ɵnov(v,1).tab.show;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10
    ]
    );
  });
}
export const TabButtonNgFactory:import0.ComponentFactory<import5.TabButton> = import0.ɵccf('.tab-button',import5.TabButton,View_TabButton_Host_0,{
  color: 'color',
  mode: 'mode',
  tab: 'tab'
}
,{ionSelect: 'ionSelect'},([] as any[]));
