// ionic-angular/components/tabs/tab.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/tabs/tab';
import * as import2 from 'ionic-angular/components/tabs/tabs';
import * as import3 from 'ionic-angular/components/app/app';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/platform/platform';
import * as import6 from 'ionic-angular/gestures/gesture-controller';
import * as import7 from 'ionic-angular/transitions/transition-controller';
import * as import8 from 'ionic-angular/navigation/deep-linker';
import * as import9 from 'ionic-angular/platform/dom-controller';
const styles_Tab:any[] = ([] as any[]);
export const RenderType_Tab:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Tab,
  data: {}
}
);
export function View_Tab_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_vp: 0}),
    (l()(),import0.ɵeld(16777216,[
      [
        1,
        3
      ]
      ,
      [
        'viewport',
        1
      ]

    ]
    ,(null as any),0,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'nav-decor'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_Tab_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-tab',[[
        'role',
        'tabpanel'
      ]
    ],[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]

    ]
    ,(null as any),(null as any),View_Tab_0,RenderType_Tab)),
    import0.ɵdid(245760,(null as any),0,import1.Tab,[
      import2.Tabs,
      import3.App,
      import4.Config,
      import5.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import0.ChangeDetectorRef,
      import6.GestureController,
      import7.TransitionController,
      [
        2,
        import8.DeepLinker
      ]
      ,
      import9.DomController,
      import0.ErrorHandler
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._tabId;
    const currVal_1:any = import0.ɵnov(v,1)._btnId;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const TabNgFactory:import0.ComponentFactory<import1.Tab> = import0.ɵccf('ion-tab',import1.Tab,View_Tab_Host_0,{
  color: 'color',
  mode: 'mode',
  swipeBackEnabled: 'swipeBackEnabled',
  root: 'root',
  rootParams: 'rootParams',
  tabUrlPath: 'tabUrlPath',
  tabTitle: 'tabTitle',
  tabIcon: 'tabIcon',
  tabBadge: 'tabBadge',
  tabBadgeStyle: 'tabBadgeStyle',
  enabled: 'enabled',
  show: 'show',
  tabsHideOnSubPages: 'tabsHideOnSubPages'
}
,{ionSelect: 'ionSelect'},([] as any[]));
