// ionic-angular/components/tabs/tabs.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './tab-button.ngfactory';
import * as import2 from 'ionic-angular/components/tabs/tab-button';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from '@angular/common';
import * as import5 from 'ionic-angular/components/tabs/tab-highlight';
import * as import6 from 'ionic-angular/platform/dom-controller';
import * as import7 from 'ionic-angular/components/tabs/tabs';
import * as import8 from 'ionic-angular/navigation/nav-controller';
import * as import9 from 'ionic-angular/navigation/view-controller';
import * as import10 from 'ionic-angular/components/app/app';
import * as import11 from 'ionic-angular/platform/platform';
import * as import12 from 'ionic-angular/navigation/deep-linker';
import * as import13 from 'ionic-angular/platform/keyboard';
import * as import14 from 'ionic-angular/components/split-pane/split-pane';
const styles_Tabs:any[] = ([] as any[]);
export const RenderType_Tabs:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Tabs,
  data: {}
}
);
function View_Tabs_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'tab-button'
      ]
      ,
      [
        'href',
        '#'
      ]
      ,
      [
        'role',
        'tab'
      ]

    ]
    ,[
      [
        1,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-controls',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]
      ,
      [
        2,
        'has-title',
        (null as any)
      ]
      ,
      [
        2,
        'has-icon',
        (null as any)
      ]
      ,
      [
        2,
        'has-title-only',
        (null as any)
      ]
      ,
      [
        2,
        'icon-only',
        (null as any)
      ]
      ,
      [
        2,
        'has-badge',
        (null as any)
      ]
      ,
      [
        2,
        'disable-hover',
        (null as any)
      ]
      ,
      [
        2,
        'tab-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'tab-hidden',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'ionSelect'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).onClick()) !== false);
        ad = (pd_0 && ad);
      }
      if (('ionSelect' === en)) {
        const pd_1:any = ((<any>co.select(v.context.$implicit)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import1.View_TabButton_0,import1.RenderType_TabButton)),
    import0.ɵdid(114688,(null as any),0,import2.TabButton,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{tab: [
        0,
        'tab'
      ]
    },{ionSelect: 'ionSelect'})
  ]
  ,(ck,v) => {
    const currVal_11:any = v.context.$implicit;
    ck(v,1,0,currVal_11);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).tab._btnId;
    const currVal_1:any = import0.ɵnov(v,1).tab._tabId;
    const currVal_2:any = import0.ɵnov(v,1).tab.isSelected;
    const currVal_3:any = import0.ɵnov(v,1).hasTitle;
    const currVal_4:any = import0.ɵnov(v,1).hasIcon;
    const currVal_5:any = import0.ɵnov(v,1).hasTitleOnly;
    const currVal_6:any = import0.ɵnov(v,1).hasIconOnly;
    const currVal_7:any = import0.ɵnov(v,1).hasBadge;
    const currVal_8:any = import0.ɵnov(v,1).disHover;
    const currVal_9:boolean = !import0.ɵnov(v,1).tab.enabled;
    const currVal_10:boolean = !import0.ɵnov(v,1).tab.show;
    ck(v,0,1,[
      currVal_0,
      currVal_1,
      currVal_2,
      currVal_3,
      currVal_4,
      currVal_5,
      currVal_6,
      currVal_7,
      currVal_8,
      currVal_9,
      currVal_10
    ]
    );
  });
}
export function View_Tabs_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_highlight: 0}),
    import0.ɵqud(402653184,2,{_tabbar: 0}),
    import0.ɵqud(402653184,3,{portal: 0}),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'tabbar',
        1
      ]

    ]
    ,(null as any),4,'div',[
      [
        'class',
        'tabbar'
      ]
      ,
      [
        'role',
        'tablist'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Tabs_1)),
    import0.ɵdid(802816,(null as any),0,import4.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'tab-highlight'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(16384,[[
        1,
        4
      ]
    ],0,import5.TabHighlight,[
      import0.ElementRef,
      import6.DomController
    ]
    ,(null as any),(null as any)),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵeld(16777216,[
      [
        3,
        3
      ]
      ,
      [
        'portal',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'tab-portal',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    var co:import7.Tabs = v.component;
    const currVal_0:any = co._tabs;
    ck(v,5,0,currVal_0);
  },(null as any));
}
function View_Tabs_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-tabs',([] as any[]),(null as any),(null as any),(null as any),View_Tabs_0,RenderType_Tabs)),
    import0.ɵdid(4374528,(null as any),0,import7.Tabs,[
      [
        2,
        import8.NavController
      ]
      ,
      [
        2,
        import9.ViewController
      ]
      ,
      import10.App,
      import3.Config,
      import0.ElementRef,
      import11.Platform,
      import0.Renderer,
      import12.DeepLinker,
      import13.Keyboard
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(6144,(null as any),import14.RootNode,(null as any),[import7.Tabs])
  ]
  ,(null as any),(null as any));
}
export const TabsNgFactory:import0.ComponentFactory<import7.Tabs> = import0.ɵccf('ion-tabs',import7.Tabs,View_Tabs_Host_0,{
  color: 'color',
  mode: 'mode',
  name: 'name',
  selectedIndex: 'selectedIndex',
  tabsLayout: 'tabsLayout',
  tabsPlacement: 'tabsPlacement',
  tabsHighlight: 'tabsHighlight'
}
,{ionChange: 'ionChange'},['*']);
