// ionic-angular/components/popover/popover-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/popover/popover-component';
import * as import2 from 'ionic-angular/components/backdrop/backdrop';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/navigation/nav-params';
import * as import5 from 'ionic-angular/navigation/view-controller';
import * as import6 from 'ionic-angular/gestures/gesture-controller';
import * as import7 from 'ionic-angular/util/module-loader';
const styles_PopoverCmp:any[] = ([] as any[]);
export const RenderType_PopoverCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_PopoverCmp,
  data: {}
}
);
export function View_PopoverCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_viewport: 0}),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'role',
        'presentation'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,[[
        8,
        'hidden',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.PopoverCmp = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co._bdClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import2.Backdrop,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'popover-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'popover-arrow'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'popover-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'popover-viewport'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(16777216,[
      [
        1,
        3
      ]
      ,
      [
        'viewport',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'nav-viewport',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.PopoverCmp = v.component;
    const currVal_0:boolean = !co.d.showBackdrop;
    ck(v,1,0,currVal_0);
  });
}
function View_PopoverCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-popover',([] as any[]),(null as any),[[
        'body',
        'keyup'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('body:keyup' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._keyUp($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_PopoverCmp_0,RenderType_PopoverCmp)),
    import0.ɵdid(180224,(null as any),0,import1.PopoverCmp,[
      import0.ComponentFactoryResolver,
      import0.ElementRef,
      import0.Renderer,
      import3.Config,
      import4.NavParams,
      import5.ViewController,
      import6.GestureController,
      import7.ModuleLoader
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const PopoverCmpNgFactory:import0.ComponentFactory<import1.PopoverCmp> = import0.ɵccf('ion-popover',import1.PopoverCmp,View_PopoverCmp_Host_0,{},{},([] as any[]));
