// ionic-angular/components/toast/toast-component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../button/button.ngfactory';
import * as import2 from 'ionic-angular/components/button/button';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from '@angular/common';
import * as import5 from 'ionic-angular/components/toast/toast-component';
import * as import6 from 'ionic-angular/navigation/view-controller';
import * as import7 from 'ionic-angular/navigation/nav-params';
const styles_ToastCmp:any[] = ([] as any[]);
export const RenderType_ToastCmp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_ToastCmp,
  data: {}
}
);
function View_ToastCmp_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'toast-message'
      ]
      ],[[
        8,
        'id',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',co.hdrId,'');
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.d.message;
    ck(v,1,0,currVal_1);
  });
}
function View_ToastCmp_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'class',
        'toast-button'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'ion-button',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.cbClick()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import1.View_Button_0,import1.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import2.Button,[
      [
        8,
        ''
      ]
      ,
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{clear: [
        0,
        'clear'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,[
      ' ',
      ' '
    ]
    ))
  ]
  ,(ck,v) => {
    const currVal_0:any = '';
    ck(v,1,0,currVal_0);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = (co.d.closeButtonText || 'Close');
    ck(v,2,0,currVal_1);
  });
}
export function View_ToastCmp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'div',[[
        'class',
        'toast-wrapper'
      ]
    ],[
      [
        2,
        'toast-bottom',
        (null as any)
      ]
      ,
      [
        2,
        'toast-middle',
        (null as any)
      ]
      ,
      [
        2,
        'toast-top',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'toast-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ToastCmp_1)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_ToastCmp_2)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵted((null as any),[' ']))
  ]
  ,(ck,v) => {
    var co:import5.ToastCmp = v.component;
    const currVal_3:any = co.d.message;
    ck(v,5,0,currVal_3);
    const currVal_4:any = co.d.showCloseButton;
    ck(v,8,0,currVal_4);
  },(ck,v) => {
    var co:import5.ToastCmp = v.component;
    const currVal_0:any = (co.d.position === 'bottom');
    const currVal_1:any = (co.d.position === 'middle');
    const currVal_2:any = (co.d.position === 'top');
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_ToastCmp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-toast',[[
        'role',
        'dialog'
      ]
    ],[
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'aria-describedby',
        0
      ]

    ]
    ,(null as any),(null as any),View_ToastCmp_0,RenderType_ToastCmp)),
    import0.ɵdid(4243456,(null as any),0,import5.ToastCmp,[
      import6.ViewController,
      import3.Config,
      import0.ElementRef,
      import7.NavParams,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).hdrId;
    const currVal_1:any = import0.ɵnov(v,1).descId;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const ToastCmpNgFactory:import0.ComponentFactory<import5.ToastCmp> = import0.ɵccf('ion-toast',import5.ToastCmp,View_ToastCmp_Host_0,{},{},([] as any[]));
