// ionic-angular/components/datetime/datetime.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '../button/button.ngfactory';
import * as import3 from 'ionic-angular/components/button/button';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/components/datetime/datetime';
import * as import6 from 'ionic-angular/util/form';
import * as import7 from 'ionic-angular/components/item/item';
import * as import8 from 'ionic-angular/components/picker/picker-controller';
import * as import9 from '@angular/forms';
const styles_DateTime:any[] = ([] as any[]);
export const RenderType_DateTime:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DateTime,
  data: {}
}
);
function View_DateTime_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'datetime-text datetime-placeholder'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.placeholder;
    ck(v,1,0,currVal_0);
  });
}
function View_DateTime_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'datetime-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co._text;
    ck(v,1,0,currVal_0);
  });
}
export function View_DateTime_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DateTime_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DateTime_2)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'aria-haspopup',
        'true'
      ]
      ,
      [
        'class',
        'item-cover'
      ]
      ,
      [
        'ion-button',
        'item-cover'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
    ,(null as any),(null as any),import2.View_Button_0,import2.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import3.Button,[
      [
        8,
        'item-cover'
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    var co:import5.DateTime = v.component;
    const currVal_0:boolean = !co._text;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co._text;
    ck(v,3,0,currVal_1);
  },(ck,v) => {
    var co:import5.DateTime = v.component;
    const currVal_2:any = co.id;
    const currVal_3:any = co._labelId;
    const currVal_4:any = co._disabled;
    ck(v,4,0,currVal_2,currVal_3,currVal_4);
  });
}
function View_DateTime_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-datetime',([] as any[]),[[
        2,
        'datetime-disabled',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'keyup.space'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._click($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('keyup.space' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1)._keyup()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_DateTime_0,RenderType_DateTime)),
    import0.ɵdid(1228800,(null as any),0,import5.DateTime,[
      import6.Form,
      import4.Config,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import7.Item
      ]
      ,
      [
        2,
        import8.PickerController
      ]

    ]
    ,(null as any),(null as any)),
    import0.ɵprd(5120,(null as any),import9.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import5.DateTime])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._disabled;
    ck(v,0,0,currVal_0);
  });
}
export const DateTimeNgFactory:import0.ComponentFactory<import5.DateTime> = import0.ɵccf('ion-datetime',import5.DateTime,View_DateTime_Host_0,{
  color: 'color',
  mode: 'mode',
  disabled: 'disabled',
  min: 'min',
  max: 'max',
  displayFormat: 'displayFormat',
  pickerFormat: 'pickerFormat',
  cancelText: 'cancelText',
  doneText: 'doneText',
  yearValues: 'yearValues',
  monthValues: 'monthValues',
  dayValues: 'dayValues',
  hourValues: 'hourValues',
  minuteValues: 'minuteValues',
  monthNames: 'monthNames',
  monthShortNames: 'monthShortNames',
  dayNames: 'dayNames',
  dayShortNames: 'dayShortNames',
  pickerOptions: 'pickerOptions',
  placeholder: 'placeholder'
}
,{
  ionFocus: 'ionFocus',
  ionChange: 'ionChange',
  ionBlur: 'ionBlur',
  ionCancel: 'ionCancel'
}
,([] as any[]));
