// ionic-angular/components/app/app-root.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/app/overlay-portal';
import * as import2 from 'ionic-angular/components/app/app';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/platform/platform';
import * as import5 from 'ionic-angular/gestures/gesture-controller';
import * as import6 from 'ionic-angular/transitions/transition-controller';
import * as import7 from 'ionic-angular/navigation/deep-linker';
import * as import8 from 'ionic-angular/platform/dom-controller';
import * as import9 from 'ionic-angular/components/app/click-block';
import * as import10 from 'ionic-angular/components/app/app-root';
const styles_IonicApp:any[] = ([] as any[]);
export const RenderType_IonicApp:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_IonicApp,
  data: {}
}
);
export function View_IonicApp_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_viewport: 0}),
    import0.ɵqud(402653184,2,{_modalPortal: 0}),
    import0.ɵqud(402653184,3,{_overlayPortal: 0}),
    import0.ɵqud(402653184,4,{_loadingPortal: 0}),
    import0.ɵqud(402653184,5,{_toastPortal: 0}),
    (l()(),import0.ɵeld(16777216,[
      [
        1,
        3
      ]
      ,
      [
        'viewport',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'app-viewport',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(16777216,[[
        'modalPortal',
        1
      ]
      ],(null as any),1,'div',[[
        'overlay-portal',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(147456,[[
        2,
        4
      ]
    ],0,import1.OverlayPortal,[
      import2.App,
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import5.GestureController,
      import6.TransitionController,
      [
        2,
        import7.DeepLinker
      ]
      ,
      import0.ViewContainerRef,
      import8.DomController,
      import0.ErrorHandler
    ]
      ,{_overlayPortal: [
        0,
        '_overlayPortal'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(16777216,[[
        'overlayPortal',
        1
      ]
      ],(null as any),1,'div',[[
        'overlay-portal',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(147456,[[
        3,
        4
      ]
    ],0,import1.OverlayPortal,[
      import2.App,
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import5.GestureController,
      import6.TransitionController,
      [
        2,
        import7.DeepLinker
      ]
      ,
      import0.ViewContainerRef,
      import8.DomController,
      import0.ErrorHandler
    ]
      ,{_overlayPortal: [
        0,
        '_overlayPortal'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(16777216,[[
        'loadingPortal',
        1
      ]
    ],(null as any),1,'div',[
      [
        'class',
        'loading-portal'
      ]
      ,
      [
        'overlay-portal',
        ''
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(147456,[[
        4,
        4
      ]
    ],0,import1.OverlayPortal,[
      import2.App,
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import5.GestureController,
      import6.TransitionController,
      [
        2,
        import7.DeepLinker
      ]
      ,
      import0.ViewContainerRef,
      import8.DomController,
      import0.ErrorHandler
    ]
      ,{_overlayPortal: [
        0,
        '_overlayPortal'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(16777216,[[
        'toastPortal',
        1
      ]
      ],(null as any),1,'div',[[
        'class',
        'toast-portal'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(147456,[[
        5,
        4
      ]
    ],0,import1.OverlayPortal,[
      import2.App,
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import5.GestureController,
      import6.TransitionController,
      [
        2,
        import7.DeepLinker
      ]
      ,
      import0.ViewContainerRef,
      import8.DomController,
      import0.ErrorHandler
    ]
      ,{_overlayPortal: [
        0,
        '_overlayPortal'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'click-block'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(16384,(null as any),0,import9.ClickBlock,[
      import2.App,
      import3.Config,
      import4.Platform,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = '';
    ck(v,7,0,currVal_0);
    const currVal_1:any = '';
    ck(v,9,0,currVal_1);
    const currVal_2:any = '';
    ck(v,11,0,currVal_2);
    const currVal_3:any = 10000;
    ck(v,13,0,currVal_3);
  },(null as any));
}
function View_IonicApp_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-app',([] as any[]),(null as any),(null as any),(null as any),View_IonicApp_0,RenderType_IonicApp)),
    import0.ɵdid(114688,(null as any),0,import10.IonicApp,[
      import10.AppRootToken,
      import0.ComponentFactoryResolver,
      import0.ElementRef,
      import0.Renderer,
      import3.Config,
      import4.Platform,
      import2.App
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const IonicAppNgFactory:import0.ComponentFactory<import10.IonicApp> = import0.ɵccf('ion-app',import10.IonicApp,View_IonicApp_Host_0,{
  color: 'color',
  mode: 'mode'
}
,{},([] as any[]));
