// ionic-angular/components/range/range.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './range-knob.ngfactory';
import * as import2 from 'ionic-angular/components/range/range-knob';
import * as import3 from '@angular/common';
import * as import4 from 'ionic-angular/components/range/range';
import * as import5 from 'ionic-angular/util/form';
import * as import6 from 'ionic-angular/tap-click/haptic';
import * as import7 from 'ionic-angular/components/item/item';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from 'ionic-angular/platform/platform';
import * as import10 from 'ionic-angular/platform/dom-controller';
import * as import11 from '@angular/forms';
const styles_Range:any[] = ([] as any[]);
export const RenderType_Range:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Range,
  data: {}
}
);
function View_Range_1(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[
      [
        'class',
        'range-tick'
      ]
      ,
      [
        'role',
        'presentation'
      ]

    ]
    ,[
      [
        4,
        'left',
        (null as any)
      ]
      ,
      [
        2,
        'range-tick-active',
        (null as any)
      ]

    ]
  ,(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    const currVal_0:any = v.context.$implicit.left;
    const currVal_1:any = v.context.$implicit.active;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_Range_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[
      [
        'class',
        'range-knob-handle'
      ]
      ,
      [
        'role',
        'slider'
      ]

    ]
    ,[
      [
        2,
        'range-knob-pressed',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-min',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-max',
        (null as any)
      ]
      ,
      [
        4,
        'left',
        (null as any)
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'ionIncrease'
      ]
      ,
      [
        (null as any),
        'ionDecrease'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._keyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('ionIncrease' === en)) {
        const pd_1:any = ((<any>co._keyChg(true,true)) !== false);
        ad = (pd_1 && ad);
      }
      if (('ionDecrease' === en)) {
        const pd_2:any = ((<any>co._keyChg(false,true)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },import1.View_RangeKnob_0,import1.RenderType_RangeKnob)),
    import0.ɵdid(49152,(null as any),0,import2.RangeKnob,([] as any[]),{
      ratio: [
        0,
        'ratio'
      ]
      ,
      pressed: [
        1,
        'pressed'
      ]
      ,
      pin: [
        2,
        'pin'
      ]
      ,
      min: [
        3,
        'min'
      ]
      ,
      max: [
        4,
        'max'
      ]
      ,
      val: [
        5,
        'val'
      ]
      ,
      disabled: [
        6,
        'disabled'
      ]
      ,
      labelId: [
        7,
        'labelId'
      ]

    }
    ,{
      ionIncrease: 'ionIncrease',
      ionDecrease: 'ionDecrease'
    }
    )
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_10:any = co._ratioB;
    const currVal_11:any = co._pressedB;
    const currVal_12:any = co._pin;
    const currVal_13:any = co._min;
    const currVal_14:any = co._max;
    const currVal_15:any = co._valB;
    const currVal_16:any = co._disabled;
    const currVal_17:any = co._labelId;
    ck(v,1,0,currVal_10,currVal_11,currVal_12,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).pressed;
    const currVal_1:any = ((import0.ɵnov(v,1).val === import0.ɵnov(v,1).min) || (import0.ɵnov(v,1).val === (undefined as any)));
    const currVal_2:any = (import0.ɵnov(v,1).val === import0.ɵnov(v,1).max);
    const currVal_3:any = import0.ɵnov(v,1)._x;
    const currVal_4:any = import0.ɵnov(v,1).val;
    const currVal_5:any = import0.ɵnov(v,1).min;
    const currVal_6:any = import0.ɵnov(v,1).max;
    const currVal_7:any = import0.ɵnov(v,1).disabled;
    const currVal_8:any = import0.ɵnov(v,1).labelId;
    const currVal_9:any = (import0.ɵnov(v,1).disabled? (0 - 1): 0);
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
export function View_Range_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_slider: 0}),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'slider',
        1
      ]

    ]
      ,(null as any),8,'div',[[
        'class',
        'range-slider'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Range_1)),
    import0.ɵdid(802816,(null as any),0,import3.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[
      [
        'class',
        'range-bar'
      ]
      ,
      [
        'role',
        'presentation'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,[[
        'bar',
        1
      ]
    ],(null as any),0,'div',[
      [
        'class',
        'range-bar range-bar-active'
      ]
      ,
      [
        'role',
        'presentation'
      ]

    ]
    ,[
      [
        4,
        'left',
        (null as any)
      ]
      ,
      [
        4,
        'right',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[
      [
        'class',
        'range-knob-handle'
      ]
      ,
      [
        'role',
        'slider'
      ]

    ]
    ,[
      [
        2,
        'range-knob-pressed',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-min',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-max',
        (null as any)
      ]
      ,
      [
        4,
        'left',
        (null as any)
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'ionIncrease'
      ]
      ,
      [
        (null as any),
        'ionDecrease'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import4.Range = v.component;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,8)._keyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('ionIncrease' === en)) {
        const pd_1:any = ((<any>co._keyChg(true,false)) !== false);
        ad = (pd_1 && ad);
      }
      if (('ionDecrease' === en)) {
        const pd_2:any = ((<any>co._keyChg(false,false)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },import1.View_RangeKnob_0,import1.RenderType_RangeKnob)),
    import0.ɵdid(49152,(null as any),0,import2.RangeKnob,([] as any[]),{
      ratio: [
        0,
        'ratio'
      ]
      ,
      pressed: [
        1,
        'pressed'
      ]
      ,
      pin: [
        2,
        'pin'
      ]
      ,
      min: [
        3,
        'min'
      ]
      ,
      max: [
        4,
        'max'
      ]
      ,
      val: [
        5,
        'val'
      ]
      ,
      disabled: [
        6,
        'disabled'
      ]
      ,
      labelId: [
        7,
        'labelId'
      ]

    }
    ,{
      ionIncrease: 'ionIncrease',
      ionDecrease: 'ionDecrease'
    }
    ),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_Range_2)),
    import0.ɵdid(16384,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    import0.ɵncd((null as any),1)
  ]
  ,(ck,v) => {
    var co:import4.Range = v.component;
    const currVal_0:any = co._ticks;
    ck(v,4,0,currVal_0);
    const currVal_13:any = co._ratioA;
    const currVal_14:any = co._pressedA;
    const currVal_15:any = co._pin;
    const currVal_16:any = co._min;
    const currVal_17:any = co._max;
    const currVal_18:any = co._valA;
    const currVal_19:any = co._disabled;
    const currVal_20:any = co._labelId;
    ck(v,8,0,currVal_13,currVal_14,currVal_15,currVal_16,currVal_17,currVal_18,currVal_19,currVal_20);
    const currVal_21:any = co._dual;
    ck(v,10,0,currVal_21);
  },(ck,v) => {
    var co:import4.Range = v.component;
    const currVal_1:any = co._barL;
    const currVal_2:any = co._barR;
    ck(v,6,0,currVal_1,currVal_2);
    const currVal_3:any = import0.ɵnov(v,8).pressed;
    const currVal_4:any = ((import0.ɵnov(v,8).val === import0.ɵnov(v,8).min) || (import0.ɵnov(v,8).val === (undefined as any)));
    const currVal_5:any = (import0.ɵnov(v,8).val === import0.ɵnov(v,8).max);
    const currVal_6:any = import0.ɵnov(v,8)._x;
    const currVal_7:any = import0.ɵnov(v,8).val;
    const currVal_8:any = import0.ɵnov(v,8).min;
    const currVal_9:any = import0.ɵnov(v,8).max;
    const currVal_10:any = import0.ɵnov(v,8).disabled;
    const currVal_11:any = import0.ɵnov(v,8).labelId;
    const currVal_12:any = (import0.ɵnov(v,8).disabled? (0 - 1): 0);
    ck(v,7,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12);
  });
}
function View_Range_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-range',([] as any[]),[
      [
        2,
        'range-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'range-pressed',
        (null as any)
      ]
      ,
      [
        2,
        'range-has-pin',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Range_0,RenderType_Range)),
    import0.ɵdid(1228800,(null as any),0,import4.Range,[
      import5.Form,
      import6.Haptic,
      [
        2,
        import7.Item
      ]
      ,
      import8.Config,
      import9.Platform,
      import0.ElementRef,
      import0.Renderer,
      import10.DomController,
      import0.ChangeDetectorRef
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(5120,(null as any),import11.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import4.Range])
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._disabled;
    const currVal_1:any = import0.ɵnov(v,1)._pressed;
    const currVal_2:any = import0.ɵnov(v,1)._pin;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const RangeNgFactory:import0.ComponentFactory<import4.Range> = import0.ɵccf('ion-range',import4.Range,View_Range_Host_0,{
  color: 'color',
  mode: 'mode',
  disabled: 'disabled',
  min: 'min',
  max: 'max',
  step: 'step',
  snaps: 'snaps',
  pin: 'pin',
  debounce: 'debounce',
  dualKnobs: 'dualKnobs'
}
,{
  ionFocus: 'ionFocus',
  ionChange: 'ionChange',
  ionBlur: 'ionBlur'
}
,[
  '[range-left]',
  '[range-right]'
]
);
