// ionic-angular/components/range/range-knob.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from 'ionic-angular/components/range/range-knob';
const styles_RangeKnob:any[] = ([] as any[]);
export const RenderType_RangeKnob:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_RangeKnob,
  data: {}
}
);
function View_RangeKnob_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[
      [
        'class',
        'range-pin'
      ]
      ,
      [
        'role',
        'presentation'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.val;
    ck(v,1,0,currVal_0);
  });
}
export function View_RangeKnob_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_RangeKnob_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[
      [
        'class',
        'range-knob'
      ]
      ,
      [
        'role',
        'presentation'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    var co:import2.RangeKnob = v.component;
    const currVal_0:any = co.pin;
    ck(v,1,0,currVal_0);
  },(null as any));
}
function View_RangeKnob_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[
      [
        'class',
        'range-knob-handle'
      ]
      ,
      [
        'role',
        'slider'
      ]

    ]
    ,[
      [
        2,
        'range-knob-pressed',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-min',
        (null as any)
      ]
      ,
      [
        2,
        'range-knob-max',
        (null as any)
      ]
      ,
      [
        4,
        'left',
        (null as any)
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        8,
        'tabIndex',
        0
      ]

    ]
      ,[[
        (null as any),
        'keydown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._keyup($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_RangeKnob_0,RenderType_RangeKnob)),
    import0.ɵdid(49152,(null as any),0,import2.RangeKnob,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).pressed;
    const currVal_1:any = ((import0.ɵnov(v,1).val === import0.ɵnov(v,1).min) || (import0.ɵnov(v,1).val === (undefined as any)));
    const currVal_2:any = (import0.ɵnov(v,1).val === import0.ɵnov(v,1).max);
    const currVal_3:any = import0.ɵnov(v,1)._x;
    const currVal_4:any = import0.ɵnov(v,1).val;
    const currVal_5:any = import0.ɵnov(v,1).min;
    const currVal_6:any = import0.ɵnov(v,1).max;
    const currVal_7:any = import0.ɵnov(v,1).disabled;
    const currVal_8:any = import0.ɵnov(v,1).labelId;
    const currVal_9:any = (import0.ɵnov(v,1).disabled? (0 - 1): 0);
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
  });
}
export const RangeKnobNgFactory:import0.ComponentFactory<import2.RangeKnob> = import0.ɵccf('.range-knob-handle',import2.RangeKnob,View_RangeKnob_Host_0,{
  ratio: 'ratio',
  pressed: 'pressed',
  pin: 'pin',
  min: 'min',
  max: 'max',
  val: 'val',
  disabled: 'disabled',
  labelId: 'labelId'
}
,{
  ionIncrease: 'ionIncrease',
  ionDecrease: 'ionDecrease'
}
,([] as any[]));
