// ionic-angular/components/menu/menu.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/backdrop/backdrop';
import * as import2 from 'ionic-angular/components/menu/menu';
import * as import3 from 'ionic-angular/components/app/menu-controller';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/platform/platform';
import * as import6 from 'ionic-angular/platform/keyboard';
import * as import7 from 'ionic-angular/gestures/gesture-controller';
import * as import8 from 'ionic-angular/platform/dom-controller';
import * as import9 from 'ionic-angular/components/app/app';
import * as import10 from 'ionic-angular/components/split-pane/split-pane';
const styles_Menu:any[] = ([] as any[]);
export const RenderType_Menu:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Menu,
  data: {}
}
);
export function View_Menu_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    import0.ɵqud(402653184,1,{backdrop: 0}),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'menu-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-backdrop',[
      [
        'disable-activated',
        ''
      ]
      ,
      [
        'role',
        'presentation'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
      import0.ɵdid(16384,[[
        1,
        4
      ]
    ],0,import1.Backdrop,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
function View_Menu_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'ion-menu',[[
        'role',
        'navigation'
      ]
    ],(null as any),(null as any),(null as any),View_Menu_0,RenderType_Menu)),
    import0.ɵdid(245760,(null as any),2,import2.Menu,[
      import3.MenuController,
      import0.ElementRef,
      import4.Config,
      import5.Platform,
      import0.Renderer,
      import6.Keyboard,
      import7.GestureController,
      import8.DomController,
      import9.App
    ]
    ,(null as any),(null as any)),
    import0.ɵqud(335544320,1,{menuContent: 0}),
    import0.ɵqud(335544320,2,{menuNav: 0}),
    import0.ɵprd(6144,(null as any),import10.RootNode,(null as any),[import2.Menu])
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const MenuNgFactory:import0.ComponentFactory<import2.Menu> = import0.ɵccf('ion-menu',import2.Menu,View_Menu_Host_0,{
  content: 'content',
  id: 'id',
  type: 'type',
  enabled: 'enabled',
  side: 'side',
  swipeEnabled: 'swipeEnabled',
  persistent: 'persistent',
  maxEdgeStart: 'maxEdgeStart'
}
,{
  ionDrag: 'ionDrag',
  ionOpen: 'ionOpen',
  ionClose: 'ionClose'
}
,['*']);
