// ionic-angular/components/infinite-scroll/infinite-scroll-content.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../spinner/spinner.ngfactory';
import * as import2 from 'ionic-angular/components/spinner/spinner';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from '@angular/common';
import * as import5 from 'ionic-angular/components/infinite-scroll/infinite-scroll-content';
import * as import6 from 'ionic-angular/components/infinite-scroll/infinite-scroll';
const styles_InfiniteScrollContent:any[] = ([] as any[]);
export const RenderType_InfiniteScrollContent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_InfiniteScrollContent,
  data: {}
}
);
function View_InfiniteScrollContent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'infinite-loading-spinner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',([] as any[]),[[
        2,
        'spinner-paused',
        (null as any)
      ]
    ],(null as any),(null as any),import1.View_Spinner_0,import1.RenderType_Spinner)),
    import0.ɵdid(114688,(null as any),0,import2.Spinner,[
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.loadingSpinner;
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2)._paused;
    ck(v,1,0,currVal_0);
  });
}
function View_InfiniteScrollContent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'infinite-loading-text'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.loadingText;
    ck(v,0,0,currVal_0);
  });
}
export function View_InfiniteScrollContent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'infinite-loading'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_InfiniteScrollContent_1)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_InfiniteScrollContent_2)),
    import0.ɵdid(16384,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import5.InfiniteScrollContent = v.component;
    const currVal_0:any = co.loadingSpinner;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.loadingText;
    ck(v,4,0,currVal_1);
  },(null as any));
}
function View_InfiniteScrollContent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-infinite-scroll-content',([] as any[]),[[
        1,
        'state',
        0
      ]
    ],(null as any),(null as any),View_InfiniteScrollContent_0,RenderType_InfiniteScrollContent)),
    import0.ɵdid(114688,(null as any),0,import5.InfiniteScrollContent,[
      import6.InfiniteScroll,
      import3.Config
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).inf.state;
    ck(v,0,0,currVal_0);
  });
}
export const InfiniteScrollContentNgFactory:import0.ComponentFactory<import5.InfiniteScrollContent> = import0.ɵccf('ion-infinite-scroll-content',import5.InfiniteScrollContent,View_InfiniteScrollContent_Host_0,{
  loadingSpinner: 'loadingSpinner',
  loadingText: 'loadingText'
}
,{},([] as any[]));
