// ionic-angular/components/fab/fab-container.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/fab/fab-container';
import * as import2 from 'ionic-angular/platform/platform';
const styles_FabContainer:any[] = ([] as any[]);
export const RenderType_FabContainer:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FabContainer,
  data: {}
}
);
export function View_FabContainer_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_FabContainer_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'ion-fab',([] as any[]),(null as any),(null as any),(null as any),View_FabContainer_0,RenderType_FabContainer)),
    import0.ɵdid(1228800,(null as any),2,import1.FabContainer,[import2.Platform],(null as any),(null as any)),
    import0.ɵqud(335544320,1,{_mainButton: 0}),
    import0.ɵqud(603979776,2,{_fabLists: 1})
  ]
  ,(null as any),(null as any));
}
export const FabContainerNgFactory:import0.ComponentFactory<import1.FabContainer> = import0.ɵccf('ion-fab',import1.FabContainer,View_FabContainer_Host_0,{},{},['*']);
