// ionic-angular/components/fab/fab.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/icon/icon';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from 'ionic-angular/components/fab/fab';
const styles_FabButton:any[] = ([] as any[]);
export const RenderType_FabButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_FabButton,
  data: {}
}
);
export function View_FabButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[
      [
        'class',
        'fab-close-icon'
      ]
      ,
      [
        'name',
        'close'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'button-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'button-effect'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    const currVal_1:any = 'close';
    ck(v,1,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._hidden;
    ck(v,0,0,currVal_0);
  });
}
function View_FabButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'ion-fab',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_FabButton_0,RenderType_FabButton)),
    import0.ɵdid(49152,(null as any),0,import3.FabButton,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const FabButtonNgFactory:import0.ComponentFactory<import3.FabButton> = import0.ɵccf('[ion-fab]',import3.FabButton,View_FabButton_Host_0,{
  color: 'color',
  mode: 'mode'
}
,{},['*']);
