// ionic-angular/components/nav/nav.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/nav/nav';
import * as import2 from 'ionic-angular/navigation/view-controller';
import * as import3 from 'ionic-angular/navigation/nav-controller';
import * as import4 from 'ionic-angular/components/app/app';
import * as import5 from 'ionic-angular/config/config';
import * as import6 from 'ionic-angular/platform/platform';
import * as import7 from 'ionic-angular/gestures/gesture-controller';
import * as import8 from 'ionic-angular/transitions/transition-controller';
import * as import9 from 'ionic-angular/navigation/deep-linker';
import * as import10 from 'ionic-angular/platform/dom-controller';
import * as import11 from 'ionic-angular/components/split-pane/split-pane';
const styles_Nav:any[] = ([] as any[]);
export const RenderType_Nav:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Nav,
  data: {}
}
);
export function View_Nav_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_vp: 0}),
    (l()(),import0.ɵeld(16777216,[
      [
        1,
        3
      ]
      ,
      [
        'viewport',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'nav-viewport',
        ''
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'nav-decor'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_Nav_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ion-nav',([] as any[]),(null as any),(null as any),(null as any),View_Nav_0,RenderType_Nav)),
    import0.ɵdid(4374528,(null as any),0,import1.Nav,[
      [
        2,
        import2.ViewController
      ]
      ,
      [
        2,
        import3.NavController
      ]
      ,
      import4.App,
      import5.Config,
      import6.Platform,
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer,
      import0.ComponentFactoryResolver,
      import7.GestureController,
      import8.TransitionController,
      [
        2,
        import9.DeepLinker
      ]
      ,
      import10.DomController,
      import0.ErrorHandler
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(6144,(null as any),import11.RootNode,(null as any),[import1.Nav])
  ]
  ,(null as any),(null as any));
}
export const NavNgFactory:import0.ComponentFactory<import1.Nav> = import0.ɵccf('ion-nav',import1.Nav,View_Nav_Host_0,{
  color: 'color',
  mode: 'mode',
  swipeBackEnabled: 'swipeBackEnabled',
  root: 'root',
  rootParams: 'rootParams',
  name: 'name'
}
,{},([] as any[]));
