// ionic-angular/components/radio/radio-button.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../button/button.ngfactory';
import * as import2 from 'ionic-angular/components/button/button';
import * as import3 from 'ionic-angular/config/config';
import * as import4 from 'ionic-angular/components/radio/radio-button';
import * as import5 from 'ionic-angular/util/form';
import * as import6 from 'ionic-angular/components/item/item';
import * as import7 from 'ionic-angular/components/radio/radio-group';
const styles_RadioButton:any[] = ([] as any[]);
export const RenderType_RadioButton:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_RadioButton,
  data: {}
}
);
export function View_RadioButton_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'div',[[
        'class',
        'radio-icon'
      ]
      ],[[
        2,
        'radio-checked',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'radio-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵted((null as any),[' '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'button',[
      [
        'class',
        'item-cover'
      ]
      ,
      [
        'ion-button',
        'item-cover'
      ]
      ,
      [
        'role',
        'radio'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-checked',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
    ,(null as any),(null as any),import1.View_Button_0,import1.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import2.Button,[
      [
        8,
        'item-cover'
      ]
      ,
      import3.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,[' ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import4.RadioButton = v.component;
    const currVal_0:any = co._checked;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.id;
    const currVal_2:any = co._checked;
    const currVal_3:any = co._labelId;
    const currVal_4:any = co._disabled;
    ck(v,5,0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_RadioButton_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-radio',([] as any[]),[[
        2,
        'radio-disabled',
        (null as any)
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._click($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_RadioButton_0,RenderType_RadioButton)),
    import0.ɵdid(245760,(null as any),0,import4.RadioButton,[
      import5.Form,
      import3.Config,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import6.Item
      ]
      ,
      [
        2,
        import7.RadioGroup
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._disabled;
    ck(v,0,0,currVal_0);
  });
}
export const RadioButtonNgFactory:import0.ComponentFactory<import4.RadioButton> = import0.ɵccf('ion-radio',import4.RadioButton,View_RadioButton_Host_0,{
  color: 'color',
  mode: 'mode',
  value: 'value',
  checked: 'checked',
  disabled: 'disabled'
}
,{ionSelect: 'ionSelect'},([] as any[]));
