// ionic-angular/components/refresher/refresher-content.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/icon/icon';
import * as import2 from 'ionic-angular/config/config';
import * as import3 from '@angular/common';
import * as import4 from '../spinner/spinner.ngfactory';
import * as import5 from 'ionic-angular/components/spinner/spinner';
import * as import6 from 'ionic-angular/components/refresher/refresher-content';
import * as import7 from 'ionic-angular/components/refresher/refresher';
const styles_RefresherContent:any[] = ([] as any[]);
export const RenderType_RefresherContent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_RefresherContent,
  data: {}
}
);
function View_RefresherContent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'refresher-pulling-icon'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-icon',[[
        'role',
        'img'
      ]
      ],[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import1.Icon,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = co.pullingIcon;
    ck(v,2,0,currVal_1);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2)._hidden;
    ck(v,1,0,currVal_0);
  });
}
function View_RefresherContent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'refresher-pulling-text'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.pullingText;
    ck(v,0,0,currVal_0);
  });
}
function View_RefresherContent_3(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'refresher-refreshing-text'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.refreshingText;
    ck(v,0,0,currVal_0);
  });
}
export function View_RefresherContent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'refresher-pulling'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_RefresherContent_1)),
    import0.ɵdid(16384,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_RefresherContent_2)),
    import0.ɵdid(16384,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'refresher-refreshing'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'refresher-refreshing-icon'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-spinner',([] as any[]),[[
        2,
        'spinner-paused',
        (null as any)
      ]
    ],(null as any),(null as any),import4.View_Spinner_0,import4.RenderType_Spinner)),
    import0.ɵdid(114688,(null as any),0,import5.Spinner,[
      import2.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any)),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_RefresherContent_3)),
    import0.ɵdid(16384,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import6.RefresherContent = v.component;
    const currVal_0:any = co.pullingIcon;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.pullingText;
    ck(v,4,0,currVal_1);
    const currVal_3:any = co.refreshingSpinner;
    ck(v,8,0,currVal_3);
    const currVal_4:any = co.refreshingText;
    ck(v,10,0,currVal_4);
  },(ck,v) => {
    const currVal_2:any = import0.ɵnov(v,8)._paused;
    ck(v,7,0,currVal_2);
  });
}
function View_RefresherContent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-refresher-content',([] as any[]),[[
        1,
        'state',
        0
      ]
    ],(null as any),(null as any),View_RefresherContent_0,RenderType_RefresherContent)),
    import0.ɵdid(114688,(null as any),0,import6.RefresherContent,[
      import7.Refresher,
      import2.Config
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).r.state;
    ck(v,0,0,currVal_0);
  });
}
export const RefresherContentNgFactory:import0.ComponentFactory<import6.RefresherContent> = import0.ɵccf('ion-refresher-content',import6.RefresherContent,View_RefresherContent_Host_0,{
  pullingIcon: 'pullingIcon',
  pullingText: 'pullingText',
  refreshingSpinner: 'refreshingSpinner',
  refreshingText: 'refreshingText'
}
,{},([] as any[]));
