// ionic-angular/components/searchbar/searchbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ionic-angular/components/searchbar/searchbar';
import * as import2 from '../button/button.ngfactory';
import * as import3 from 'ionic-angular/components/button/button';
import * as import4 from 'ionic-angular/config/config';
import * as import5 from 'ionic-angular/components/icon/icon';
import * as import6 from 'ionic-angular/platform/platform';
import * as import7 from '@angular/forms';
const styles_Searchbar:any[] = ([] as any[]);
export const RenderType_Searchbar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_Searchbar,
  data: {}
}
);
export function View_Searchbar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(402653184,1,{_searchbarInput: 0}),
    import0.ɵqud(402653184,2,{_searchbarIcon: 0}),
    import0.ɵqud(402653184,3,{_cancelButton: 0}),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'searchbar-input-container'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),3,'button',[
      [
        'class',
        'searchbar-md-cancel'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'color',
        'dark'
      ]
      ,
      [
        'ion-button',
        ''
      ]
      ,
      [
        'mode',
        'md'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,(null as any),[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Searchbar = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.cancelSearchbar($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.cancelSearchbar($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import2.View_Button_0,import2.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import3.Button,[
      [
        8,
        ''
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      color: [
        0,
        'color'
      ]
      ,
      mode: [
        1,
        'mode'
      ]
      ,
      clear: [
        2,
        'clear'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵeld(0,(null as any),0,1,'ion-icon',[
      [
        'name',
        'md-arrow-back'
      ]
      ,
      [
        'role',
        'img'
      ]

    ]
      ,[[
        2,
        'hide',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(147456,(null as any),0,import5.Icon,[
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{name: [
        0,
        'name'
      ]
    },(null as any)),
    (l()(),import0.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'searchbarIcon',
        1
      ]

    ]
      ,(null as any),0,'div',[[
        'class',
        'searchbar-search-icon'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'searchbarInput',
        1
      ]

    ]
      ,(null as any),0,'input',[[
        'class',
        'searchbar-input'
      ]
    ],[
      [
        1,
        'placeholder',
        0
      ]
      ,
      [
        1,
        'type',
        0
      ]
      ,
      [
        1,
        'autocomplete',
        0
      ]
      ,
      [
        1,
        'autocorrect',
        0
      ]
      ,
      [
        1,
        'spellcheck',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'input'
      ]
      ,
      [
        (null as any),
        'blur'
      ]
      ,
      [
        (null as any),
        'focus'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Searchbar = v.component;
      if (('input' === en)) {
        const pd_0:any = ((<any>co.inputChanged($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>co.inputBlurred()) !== false);
        ad = (pd_1 && ad);
      }
      if (('focus' === en)) {
        const pd_2:any = ((<any>co.inputFocused()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'button',[
      [
        'class',
        'searchbar-clear-icon'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'ion-button',
        ''
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
    ,(null as any),[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Searchbar = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.clearInput($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.clearInput($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import2.View_Button_0,import2.RenderType_Button)),
    import0.ɵdid(1097728,(null as any),0,import3.Button,[
      [
        8,
        ''
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      mode: [
        0,
        'mode'
      ]
      ,
      clear: [
        1,
        'clear'
      ]

    }
    ,(null as any)),
      (l()(),import0.ɵeld(0,[[
        3,
        0
      ]
    ],(null as any),2,'button',[
      [
        'class',
        'searchbar-ios-cancel'
      ]
      ,
      [
        'clear',
        ''
      ]
      ,
      [
        'ion-button',
        ''
      ]
      ,
      [
        'mode',
        'ios'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,[[
        8,
        'tabIndex',
        0
      ]
    ],[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mousedown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import1.Searchbar = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.cancelSearchbar($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mousedown' === en)) {
        const pd_1:any = ((<any>co.cancelSearchbar($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import2.View_Button_0,import2.RenderType_Button)),
      import0.ɵdid(1097728,[[
        'cancelButton',
        4
      ]
    ],0,import3.Button,[
      [
        8,
        ''
      ]
      ,
      import4.Config,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      mode: [
        0,
        'mode'
      ]
      ,
      clear: [
        1,
        'clear'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,[
      '',
      ''
    ]
    ))
  ]
  ,(ck,v) => {
    var co:import1.Searchbar = v.component;
    const currVal_0:any = 'dark';
    const currVal_1:any = 'md';
    const currVal_2:any = '';
    ck(v,5,0,currVal_0,currVal_1,currVal_2);
    const currVal_4:any = 'md-arrow-back';
    ck(v,7,0,currVal_4);
    const currVal_10:any = co._mode;
    const currVal_11:any = '';
    ck(v,11,0,currVal_10,currVal_11);
    const currVal_13:any = 'ios';
    const currVal_14:any = '';
    ck(v,13,0,currVal_13,currVal_14);
  },(ck,v) => {
    var co:import1.Searchbar = v.component;
    const currVal_3:any = import0.ɵnov(v,7)._hidden;
    ck(v,6,0,currVal_3);
    const currVal_5:any = co.placeholder;
    const currVal_6:any = co.type;
    const currVal_7:any = co._autocomplete;
    const currVal_8:any = co._autocorrect;
    const currVal_9:any = co._spellcheck;
    ck(v,9,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9);
    const currVal_12:any = (co._isActive? 1: (0 - 1));
    ck(v,12,0,currVal_12);
    const currVal_15:any = co.cancelButtonText;
    ck(v,14,0,currVal_15);
  });
}
function View_Searchbar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ion-searchbar',([] as any[]),[
      [
        2,
        'searchbar-animated',
        (null as any)
      ]
      ,
      [
        2,
        'searchbar-has-value',
        (null as any)
      ]
      ,
      [
        2,
        'searchbar-active',
        (null as any)
      ]
      ,
      [
        2,
        'searchbar-show-cancel',
        (null as any)
      ]
      ,
      [
        2,
        'searchbar-left-aligned',
        (null as any)
      ]
      ,
      [
        2,
        'searchbar-has-focus',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_Searchbar_0,RenderType_Searchbar)),
    import0.ɵdid(1294336,(null as any),0,import1.Searchbar,[
      import4.Config,
      import6.Platform,
      import0.ElementRef,
      import0.Renderer,
      [
        2,
        import7.NgControl
      ]

    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._animated;
    const currVal_1:any = import0.ɵnov(v,1)._value;
    const currVal_2:any = import0.ɵnov(v,1)._isActive;
    const currVal_3:any = import0.ɵnov(v,1)._showCancelButton;
    const currVal_4:any = import0.ɵnov(v,1)._shouldAlignLeft;
    const currVal_5:any = import0.ɵnov(v,1)._isFocus;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
  });
}
export const SearchbarNgFactory:import0.ComponentFactory<import1.Searchbar> = import0.ɵccf('ion-searchbar',import1.Searchbar,View_Searchbar_Host_0,{
  color: 'color',
  mode: 'mode',
  disabled: 'disabled',
  cancelButtonText: 'cancelButtonText',
  showCancelButton: 'showCancelButton',
  debounce: 'debounce',
  placeholder: 'placeholder',
  autocomplete: 'autocomplete',
  autocorrect: 'autocorrect',
  spellcheck: 'spellcheck',
  type: 'type',
  animated: 'animated'
}
,{
  ionFocus: 'ionFocus',
  ionChange: 'ionChange',
  ionBlur: 'ionBlur',
  ionInput: 'ionInput',
  ionCancel: 'ionCancel',
  ionClear: 'ionClear'
}
,([] as any[]));
